// proto-ts-common/incoterm_pb.js
// source: bigblue/common/incoterm.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global =
    (typeof globalThis !== 'undefined' && globalThis) ||
    (typeof window !== 'undefined' && window) ||
    (typeof global !== 'undefined' && global) ||
    (typeof self !== 'undefined' && self) ||
    (function () { return this; }).call(null) ||
    Function('return this')();

goog.exportSymbol('proto.bigblue.common.Incoterm', null, global);
/**
 * @enum {number}
 */
proto.bigblue.common.Incoterm = {
  INCOTERM_UNSPECIFIED: 0,
  CPT: 1,
  CPI: 2,
  CFR: 3,
  CIF: 4,
  DAP: 5,
  DAF: 6,
  DDU: 7,
  DES: 8,
  FAS: 9,
  FCA: 10,
  DAT: 11,
  FOB: 12,
  EXW: 13,
  DPU: 14,
  DDP: 15
};

goog.object.extend(exports, proto.bigblue.common);
