// proto-ts-common/shops_pb.js
// source: bigblue/common/shops.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global =
    (typeof globalThis !== 'undefined' && globalThis) ||
    (typeof window !== 'undefined' && window) ||
    (typeof global !== 'undefined' && global) ||
    (typeof self !== 'undefined' && self) ||
    (function () { return this; }).call(null) ||
    Function('return this')();

goog.exportSymbol('proto.bigblue.common.ProcessReportStatus', null, global);
goog.exportSymbol('proto.bigblue.common.ShopOrderProcessReport', null, global);
goog.exportSymbol('proto.bigblue.common.ShopOrderProductError', null, global);
goog.exportSymbol('proto.bigblue.common.ShopOrderProductError.Type', null, global);
goog.exportSymbol('proto.bigblue.common.ShopOrdersFilter', null, global);
goog.exportSymbol('proto.bigblue.common.ShopProductsFilter', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.common.ShopOrderProcessReport = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.bigblue.common.ShopOrderProcessReport.repeatedFields_, null);
};
goog.inherits(proto.bigblue.common.ShopOrderProcessReport, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.common.ShopOrderProcessReport.displayName = 'proto.bigblue.common.ShopOrderProcessReport';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.common.ShopOrderProductError = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.bigblue.common.ShopOrderProductError, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.common.ShopOrderProductError.displayName = 'proto.bigblue.common.ShopOrderProductError';
}

/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.bigblue.common.ShopOrderProcessReport.repeatedFields_ = [2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.common.ShopOrderProcessReport.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.common.ShopOrderProcessReport.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.common.ShopOrderProcessReport} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.common.ShopOrderProcessReport.toObject = function(includeInstance, msg) {
  var f, obj = {
    errorMessage: jspb.Message.getFieldWithDefault(msg, 1, ""),
    productErrorsList: jspb.Message.toObjectList(msg.getProductErrorsList(),
    proto.bigblue.common.ShopOrderProductError.toObject, includeInstance),
    ignoreReason: jspb.Message.getFieldWithDefault(msg, 3, ""),
    status: jspb.Message.getFieldWithDefault(msg, 4, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.common.ShopOrderProcessReport}
 */
proto.bigblue.common.ShopOrderProcessReport.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.common.ShopOrderProcessReport;
  return proto.bigblue.common.ShopOrderProcessReport.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.common.ShopOrderProcessReport} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.common.ShopOrderProcessReport}
 */
proto.bigblue.common.ShopOrderProcessReport.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setErrorMessage(value);
      break;
    case 2:
      var value = new proto.bigblue.common.ShopOrderProductError;
      reader.readMessage(value,proto.bigblue.common.ShopOrderProductError.deserializeBinaryFromReader);
      msg.addProductErrors(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setIgnoreReason(value);
      break;
    case 4:
      var value = /** @type {!proto.bigblue.common.ProcessReportStatus} */ (reader.readEnum());
      msg.setStatus(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.common.ShopOrderProcessReport.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.common.ShopOrderProcessReport.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.common.ShopOrderProcessReport} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.common.ShopOrderProcessReport.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getErrorMessage();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getProductErrorsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      2,
      f,
      proto.bigblue.common.ShopOrderProductError.serializeBinaryToWriter
    );
  }
  f = message.getIgnoreReason();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getStatus();
  if (f !== 0.0) {
    writer.writeEnum(
      4,
      f
    );
  }
};


/**
 * optional string error_message = 1;
 * @return {string}
 */
proto.bigblue.common.ShopOrderProcessReport.prototype.getErrorMessage = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.common.ShopOrderProcessReport} returns this
 */
proto.bigblue.common.ShopOrderProcessReport.prototype.setErrorMessage = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * repeated ShopOrderProductError product_errors = 2;
 * @return {!Array<!proto.bigblue.common.ShopOrderProductError>}
 */
proto.bigblue.common.ShopOrderProcessReport.prototype.getProductErrorsList = function() {
  return /** @type{!Array<!proto.bigblue.common.ShopOrderProductError>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.bigblue.common.ShopOrderProductError, 2));
};


/**
 * @param {!Array<!proto.bigblue.common.ShopOrderProductError>} value
 * @return {!proto.bigblue.common.ShopOrderProcessReport} returns this
*/
proto.bigblue.common.ShopOrderProcessReport.prototype.setProductErrorsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 2, value);
};


/**
 * @param {!proto.bigblue.common.ShopOrderProductError=} opt_value
 * @param {number=} opt_index
 * @return {!proto.bigblue.common.ShopOrderProductError}
 */
proto.bigblue.common.ShopOrderProcessReport.prototype.addProductErrors = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 2, opt_value, proto.bigblue.common.ShopOrderProductError, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.bigblue.common.ShopOrderProcessReport} returns this
 */
proto.bigblue.common.ShopOrderProcessReport.prototype.clearProductErrorsList = function() {
  return this.setProductErrorsList([]);
};


/**
 * optional string ignore_reason = 3;
 * @return {string}
 */
proto.bigblue.common.ShopOrderProcessReport.prototype.getIgnoreReason = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.common.ShopOrderProcessReport} returns this
 */
proto.bigblue.common.ShopOrderProcessReport.prototype.setIgnoreReason = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional ProcessReportStatus status = 4;
 * @return {!proto.bigblue.common.ProcessReportStatus}
 */
proto.bigblue.common.ShopOrderProcessReport.prototype.getStatus = function() {
  return /** @type {!proto.bigblue.common.ProcessReportStatus} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {!proto.bigblue.common.ProcessReportStatus} value
 * @return {!proto.bigblue.common.ShopOrderProcessReport} returns this
 */
proto.bigblue.common.ShopOrderProcessReport.prototype.setStatus = function(value) {
  return jspb.Message.setProto3EnumField(this, 4, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.common.ShopOrderProductError.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.common.ShopOrderProductError.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.common.ShopOrderProductError} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.common.ShopOrderProductError.toObject = function(includeInstance, msg) {
  var f, obj = {
    type: jspb.Message.getFieldWithDefault(msg, 1, 0),
    product: jspb.Message.getFieldWithDefault(msg, 2, ""),
    productShopUrl: jspb.Message.getFieldWithDefault(msg, 3, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.common.ShopOrderProductError}
 */
proto.bigblue.common.ShopOrderProductError.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.common.ShopOrderProductError;
  return proto.bigblue.common.ShopOrderProductError.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.common.ShopOrderProductError} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.common.ShopOrderProductError}
 */
proto.bigblue.common.ShopOrderProductError.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {!proto.bigblue.common.ShopOrderProductError.Type} */ (reader.readEnum());
      msg.setType(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setProduct(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setProductShopUrl(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.common.ShopOrderProductError.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.common.ShopOrderProductError.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.common.ShopOrderProductError} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.common.ShopOrderProductError.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getType();
  if (f !== 0.0) {
    writer.writeEnum(
      1,
      f
    );
  }
  f = message.getProduct();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getProductShopUrl();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
};


/**
 * @enum {number}
 */
proto.bigblue.common.ShopOrderProductError.Type = {
  TYPE_UNSPECIFIED: 0,
  NOT_FOUND: 1,
  NOT_LINKED: 2,
  NOT_MANAGED: 3
};

/**
 * optional Type type = 1;
 * @return {!proto.bigblue.common.ShopOrderProductError.Type}
 */
proto.bigblue.common.ShopOrderProductError.prototype.getType = function() {
  return /** @type {!proto.bigblue.common.ShopOrderProductError.Type} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {!proto.bigblue.common.ShopOrderProductError.Type} value
 * @return {!proto.bigblue.common.ShopOrderProductError} returns this
 */
proto.bigblue.common.ShopOrderProductError.prototype.setType = function(value) {
  return jspb.Message.setProto3EnumField(this, 1, value);
};


/**
 * optional string product = 2;
 * @return {string}
 */
proto.bigblue.common.ShopOrderProductError.prototype.getProduct = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.common.ShopOrderProductError} returns this
 */
proto.bigblue.common.ShopOrderProductError.prototype.setProduct = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string product_shop_url = 3;
 * @return {string}
 */
proto.bigblue.common.ShopOrderProductError.prototype.getProductShopUrl = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.common.ShopOrderProductError} returns this
 */
proto.bigblue.common.ShopOrderProductError.prototype.setProductShopUrl = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * @enum {number}
 */
proto.bigblue.common.ProcessReportStatus = {
  SUCCESS: 0,
  FAILURE: 1,
  IGNORED: 2
};

/**
 * @enum {number}
 */
proto.bigblue.common.ShopProductsFilter = {
  PRODUCTS_FILTER_UNSPECIFIED: 0,
  PRODUCTS_MANAGED: 1,
  PRODUCTS_UNMANAGED: 2
};

/**
 * @enum {number}
 */
proto.bigblue.common.ShopOrdersFilter = {
  ORDERS_FILTER_UNSPECIFIED: 0,
  ORDERS_WITH_ERROR: 1
};

goog.object.extend(exports, proto.bigblue.common);
