// proto-ts-products/v0/event_pb.js
// source: bigblue/products/v0/event.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global =
    (typeof globalThis !== 'undefined' && globalThis) ||
    (typeof window !== 'undefined' && window) ||
    (typeof global !== 'undefined' && global) ||
    (typeof self !== 'undefined' && self) ||
    (function () { return this; }).call(null) ||
    Function('return this')();

var bigblue_products_v0_container_pb = require('./container_pb.js');
goog.object.extend(proto, bigblue_products_v0_container_pb);
var bigblue_products_v0_packunit_pb = require('./packunit_pb.js');
goog.object.extend(proto, bigblue_products_v0_packunit_pb);
var bigblue_products_v0_product_pb = require('./product_pb.js');
goog.object.extend(proto, bigblue_products_v0_product_pb);
goog.exportSymbol('proto.bigblue.products.v0.BundleDeleted', null, global);
goog.exportSymbol('proto.bigblue.products.v0.BundleUpserted', null, global);
goog.exportSymbol('proto.bigblue.products.v0.ContainerDeleted', null, global);
goog.exportSymbol('proto.bigblue.products.v0.ContainerUpserted', null, global);
goog.exportSymbol('proto.bigblue.products.v0.ProductDeleted', null, global);
goog.exportSymbol('proto.bigblue.products.v0.ProductPackUnitDeleted', null, global);
goog.exportSymbol('proto.bigblue.products.v0.ProductPackUnitUpserted', null, global);
goog.exportSymbol('proto.bigblue.products.v0.ProductUpserted', null, global);
goog.exportSymbol('proto.bigblue.products.v0.ProductsEvent', null, global);
goog.exportSymbol('proto.bigblue.products.v0.ProductsEvent.EventCase', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.ProductsEvent = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, proto.bigblue.products.v0.ProductsEvent.oneofGroups_);
};
goog.inherits(proto.bigblue.products.v0.ProductsEvent, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.ProductsEvent.displayName = 'proto.bigblue.products.v0.ProductsEvent';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.ProductUpserted = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.bigblue.products.v0.ProductUpserted, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.ProductUpserted.displayName = 'proto.bigblue.products.v0.ProductUpserted';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.ProductDeleted = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.bigblue.products.v0.ProductDeleted, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.ProductDeleted.displayName = 'proto.bigblue.products.v0.ProductDeleted';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.BundleUpserted = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.bigblue.products.v0.BundleUpserted, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.BundleUpserted.displayName = 'proto.bigblue.products.v0.BundleUpserted';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.BundleDeleted = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.bigblue.products.v0.BundleDeleted, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.BundleDeleted.displayName = 'proto.bigblue.products.v0.BundleDeleted';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.ContainerUpserted = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.bigblue.products.v0.ContainerUpserted, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.ContainerUpserted.displayName = 'proto.bigblue.products.v0.ContainerUpserted';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.ContainerDeleted = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.bigblue.products.v0.ContainerDeleted, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.ContainerDeleted.displayName = 'proto.bigblue.products.v0.ContainerDeleted';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.ProductPackUnitUpserted = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.bigblue.products.v0.ProductPackUnitUpserted, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.ProductPackUnitUpserted.displayName = 'proto.bigblue.products.v0.ProductPackUnitUpserted';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.ProductPackUnitDeleted = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.bigblue.products.v0.ProductPackUnitDeleted, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.ProductPackUnitDeleted.displayName = 'proto.bigblue.products.v0.ProductPackUnitDeleted';
}

/**
 * Oneof group definitions for this message. Each group defines the field
 * numbers belonging to that group. When of these fields' value is set, all
 * other fields in the group are cleared. During deserialization, if multiple
 * fields are encountered for a group, only the last value seen will be kept.
 * @private {!Array<!Array<number>>}
 * @const
 */
proto.bigblue.products.v0.ProductsEvent.oneofGroups_ = [[1,2,3,4,5,6,7,8]];

/**
 * @enum {number}
 */
proto.bigblue.products.v0.ProductsEvent.EventCase = {
  EVENT_NOT_SET: 0,
  PRODUCT_UPSERTED: 1,
  PRODUCT_DELETED: 2,
  BUNDLE_UPSERTED: 3,
  BUNDLE_DELETED: 4,
  CONTAINER_UPSERTED: 5,
  CONTAINER_DELETED: 6,
  PRODUCT_PACK_UNIT_UPSERTED: 7,
  PRODUCT_PACK_UNIT_DELETED: 8
};

/**
 * @return {proto.bigblue.products.v0.ProductsEvent.EventCase}
 */
proto.bigblue.products.v0.ProductsEvent.prototype.getEventCase = function() {
  return /** @type {proto.bigblue.products.v0.ProductsEvent.EventCase} */(jspb.Message.computeOneofCase(this, proto.bigblue.products.v0.ProductsEvent.oneofGroups_[0]));
};



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.ProductsEvent.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.ProductsEvent.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.ProductsEvent} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductsEvent.toObject = function(includeInstance, msg) {
  var f, obj = {
    productUpserted: (f = msg.getProductUpserted()) && proto.bigblue.products.v0.ProductUpserted.toObject(includeInstance, f),
    productDeleted: (f = msg.getProductDeleted()) && proto.bigblue.products.v0.ProductDeleted.toObject(includeInstance, f),
    bundleUpserted: (f = msg.getBundleUpserted()) && proto.bigblue.products.v0.BundleUpserted.toObject(includeInstance, f),
    bundleDeleted: (f = msg.getBundleDeleted()) && proto.bigblue.products.v0.BundleDeleted.toObject(includeInstance, f),
    containerUpserted: (f = msg.getContainerUpserted()) && proto.bigblue.products.v0.ContainerUpserted.toObject(includeInstance, f),
    containerDeleted: (f = msg.getContainerDeleted()) && proto.bigblue.products.v0.ContainerDeleted.toObject(includeInstance, f),
    productPackUnitUpserted: (f = msg.getProductPackUnitUpserted()) && proto.bigblue.products.v0.ProductPackUnitUpserted.toObject(includeInstance, f),
    productPackUnitDeleted: (f = msg.getProductPackUnitDeleted()) && proto.bigblue.products.v0.ProductPackUnitDeleted.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.ProductsEvent}
 */
proto.bigblue.products.v0.ProductsEvent.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.ProductsEvent;
  return proto.bigblue.products.v0.ProductsEvent.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.ProductsEvent} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.ProductsEvent}
 */
proto.bigblue.products.v0.ProductsEvent.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new proto.bigblue.products.v0.ProductUpserted;
      reader.readMessage(value,proto.bigblue.products.v0.ProductUpserted.deserializeBinaryFromReader);
      msg.setProductUpserted(value);
      break;
    case 2:
      var value = new proto.bigblue.products.v0.ProductDeleted;
      reader.readMessage(value,proto.bigblue.products.v0.ProductDeleted.deserializeBinaryFromReader);
      msg.setProductDeleted(value);
      break;
    case 3:
      var value = new proto.bigblue.products.v0.BundleUpserted;
      reader.readMessage(value,proto.bigblue.products.v0.BundleUpserted.deserializeBinaryFromReader);
      msg.setBundleUpserted(value);
      break;
    case 4:
      var value = new proto.bigblue.products.v0.BundleDeleted;
      reader.readMessage(value,proto.bigblue.products.v0.BundleDeleted.deserializeBinaryFromReader);
      msg.setBundleDeleted(value);
      break;
    case 5:
      var value = new proto.bigblue.products.v0.ContainerUpserted;
      reader.readMessage(value,proto.bigblue.products.v0.ContainerUpserted.deserializeBinaryFromReader);
      msg.setContainerUpserted(value);
      break;
    case 6:
      var value = new proto.bigblue.products.v0.ContainerDeleted;
      reader.readMessage(value,proto.bigblue.products.v0.ContainerDeleted.deserializeBinaryFromReader);
      msg.setContainerDeleted(value);
      break;
    case 7:
      var value = new proto.bigblue.products.v0.ProductPackUnitUpserted;
      reader.readMessage(value,proto.bigblue.products.v0.ProductPackUnitUpserted.deserializeBinaryFromReader);
      msg.setProductPackUnitUpserted(value);
      break;
    case 8:
      var value = new proto.bigblue.products.v0.ProductPackUnitDeleted;
      reader.readMessage(value,proto.bigblue.products.v0.ProductPackUnitDeleted.deserializeBinaryFromReader);
      msg.setProductPackUnitDeleted(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.ProductsEvent.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.ProductsEvent.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.ProductsEvent} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductsEvent.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getProductUpserted();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      proto.bigblue.products.v0.ProductUpserted.serializeBinaryToWriter
    );
  }
  f = message.getProductDeleted();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      proto.bigblue.products.v0.ProductDeleted.serializeBinaryToWriter
    );
  }
  f = message.getBundleUpserted();
  if (f != null) {
    writer.writeMessage(
      3,
      f,
      proto.bigblue.products.v0.BundleUpserted.serializeBinaryToWriter
    );
  }
  f = message.getBundleDeleted();
  if (f != null) {
    writer.writeMessage(
      4,
      f,
      proto.bigblue.products.v0.BundleDeleted.serializeBinaryToWriter
    );
  }
  f = message.getContainerUpserted();
  if (f != null) {
    writer.writeMessage(
      5,
      f,
      proto.bigblue.products.v0.ContainerUpserted.serializeBinaryToWriter
    );
  }
  f = message.getContainerDeleted();
  if (f != null) {
    writer.writeMessage(
      6,
      f,
      proto.bigblue.products.v0.ContainerDeleted.serializeBinaryToWriter
    );
  }
  f = message.getProductPackUnitUpserted();
  if (f != null) {
    writer.writeMessage(
      7,
      f,
      proto.bigblue.products.v0.ProductPackUnitUpserted.serializeBinaryToWriter
    );
  }
  f = message.getProductPackUnitDeleted();
  if (f != null) {
    writer.writeMessage(
      8,
      f,
      proto.bigblue.products.v0.ProductPackUnitDeleted.serializeBinaryToWriter
    );
  }
};


/**
 * optional ProductUpserted product_upserted = 1;
 * @return {?proto.bigblue.products.v0.ProductUpserted}
 */
proto.bigblue.products.v0.ProductsEvent.prototype.getProductUpserted = function() {
  return /** @type{?proto.bigblue.products.v0.ProductUpserted} */ (
    jspb.Message.getWrapperField(this, proto.bigblue.products.v0.ProductUpserted, 1));
};


/**
 * @param {?proto.bigblue.products.v0.ProductUpserted|undefined} value
 * @return {!proto.bigblue.products.v0.ProductsEvent} returns this
*/
proto.bigblue.products.v0.ProductsEvent.prototype.setProductUpserted = function(value) {
  return jspb.Message.setOneofWrapperField(this, 1, proto.bigblue.products.v0.ProductsEvent.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.ProductsEvent} returns this
 */
proto.bigblue.products.v0.ProductsEvent.prototype.clearProductUpserted = function() {
  return this.setProductUpserted(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.ProductsEvent.prototype.hasProductUpserted = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional ProductDeleted product_deleted = 2;
 * @return {?proto.bigblue.products.v0.ProductDeleted}
 */
proto.bigblue.products.v0.ProductsEvent.prototype.getProductDeleted = function() {
  return /** @type{?proto.bigblue.products.v0.ProductDeleted} */ (
    jspb.Message.getWrapperField(this, proto.bigblue.products.v0.ProductDeleted, 2));
};


/**
 * @param {?proto.bigblue.products.v0.ProductDeleted|undefined} value
 * @return {!proto.bigblue.products.v0.ProductsEvent} returns this
*/
proto.bigblue.products.v0.ProductsEvent.prototype.setProductDeleted = function(value) {
  return jspb.Message.setOneofWrapperField(this, 2, proto.bigblue.products.v0.ProductsEvent.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.ProductsEvent} returns this
 */
proto.bigblue.products.v0.ProductsEvent.prototype.clearProductDeleted = function() {
  return this.setProductDeleted(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.ProductsEvent.prototype.hasProductDeleted = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * optional BundleUpserted bundle_upserted = 3;
 * @return {?proto.bigblue.products.v0.BundleUpserted}
 */
proto.bigblue.products.v0.ProductsEvent.prototype.getBundleUpserted = function() {
  return /** @type{?proto.bigblue.products.v0.BundleUpserted} */ (
    jspb.Message.getWrapperField(this, proto.bigblue.products.v0.BundleUpserted, 3));
};


/**
 * @param {?proto.bigblue.products.v0.BundleUpserted|undefined} value
 * @return {!proto.bigblue.products.v0.ProductsEvent} returns this
*/
proto.bigblue.products.v0.ProductsEvent.prototype.setBundleUpserted = function(value) {
  return jspb.Message.setOneofWrapperField(this, 3, proto.bigblue.products.v0.ProductsEvent.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.ProductsEvent} returns this
 */
proto.bigblue.products.v0.ProductsEvent.prototype.clearBundleUpserted = function() {
  return this.setBundleUpserted(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.ProductsEvent.prototype.hasBundleUpserted = function() {
  return jspb.Message.getField(this, 3) != null;
};


/**
 * optional BundleDeleted bundle_deleted = 4;
 * @return {?proto.bigblue.products.v0.BundleDeleted}
 */
proto.bigblue.products.v0.ProductsEvent.prototype.getBundleDeleted = function() {
  return /** @type{?proto.bigblue.products.v0.BundleDeleted} */ (
    jspb.Message.getWrapperField(this, proto.bigblue.products.v0.BundleDeleted, 4));
};


/**
 * @param {?proto.bigblue.products.v0.BundleDeleted|undefined} value
 * @return {!proto.bigblue.products.v0.ProductsEvent} returns this
*/
proto.bigblue.products.v0.ProductsEvent.prototype.setBundleDeleted = function(value) {
  return jspb.Message.setOneofWrapperField(this, 4, proto.bigblue.products.v0.ProductsEvent.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.ProductsEvent} returns this
 */
proto.bigblue.products.v0.ProductsEvent.prototype.clearBundleDeleted = function() {
  return this.setBundleDeleted(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.ProductsEvent.prototype.hasBundleDeleted = function() {
  return jspb.Message.getField(this, 4) != null;
};


/**
 * optional ContainerUpserted container_upserted = 5;
 * @return {?proto.bigblue.products.v0.ContainerUpserted}
 */
proto.bigblue.products.v0.ProductsEvent.prototype.getContainerUpserted = function() {
  return /** @type{?proto.bigblue.products.v0.ContainerUpserted} */ (
    jspb.Message.getWrapperField(this, proto.bigblue.products.v0.ContainerUpserted, 5));
};


/**
 * @param {?proto.bigblue.products.v0.ContainerUpserted|undefined} value
 * @return {!proto.bigblue.products.v0.ProductsEvent} returns this
*/
proto.bigblue.products.v0.ProductsEvent.prototype.setContainerUpserted = function(value) {
  return jspb.Message.setOneofWrapperField(this, 5, proto.bigblue.products.v0.ProductsEvent.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.ProductsEvent} returns this
 */
proto.bigblue.products.v0.ProductsEvent.prototype.clearContainerUpserted = function() {
  return this.setContainerUpserted(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.ProductsEvent.prototype.hasContainerUpserted = function() {
  return jspb.Message.getField(this, 5) != null;
};


/**
 * optional ContainerDeleted container_deleted = 6;
 * @return {?proto.bigblue.products.v0.ContainerDeleted}
 */
proto.bigblue.products.v0.ProductsEvent.prototype.getContainerDeleted = function() {
  return /** @type{?proto.bigblue.products.v0.ContainerDeleted} */ (
    jspb.Message.getWrapperField(this, proto.bigblue.products.v0.ContainerDeleted, 6));
};


/**
 * @param {?proto.bigblue.products.v0.ContainerDeleted|undefined} value
 * @return {!proto.bigblue.products.v0.ProductsEvent} returns this
*/
proto.bigblue.products.v0.ProductsEvent.prototype.setContainerDeleted = function(value) {
  return jspb.Message.setOneofWrapperField(this, 6, proto.bigblue.products.v0.ProductsEvent.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.ProductsEvent} returns this
 */
proto.bigblue.products.v0.ProductsEvent.prototype.clearContainerDeleted = function() {
  return this.setContainerDeleted(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.ProductsEvent.prototype.hasContainerDeleted = function() {
  return jspb.Message.getField(this, 6) != null;
};


/**
 * optional ProductPackUnitUpserted product_pack_unit_upserted = 7;
 * @return {?proto.bigblue.products.v0.ProductPackUnitUpserted}
 */
proto.bigblue.products.v0.ProductsEvent.prototype.getProductPackUnitUpserted = function() {
  return /** @type{?proto.bigblue.products.v0.ProductPackUnitUpserted} */ (
    jspb.Message.getWrapperField(this, proto.bigblue.products.v0.ProductPackUnitUpserted, 7));
};


/**
 * @param {?proto.bigblue.products.v0.ProductPackUnitUpserted|undefined} value
 * @return {!proto.bigblue.products.v0.ProductsEvent} returns this
*/
proto.bigblue.products.v0.ProductsEvent.prototype.setProductPackUnitUpserted = function(value) {
  return jspb.Message.setOneofWrapperField(this, 7, proto.bigblue.products.v0.ProductsEvent.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.ProductsEvent} returns this
 */
proto.bigblue.products.v0.ProductsEvent.prototype.clearProductPackUnitUpserted = function() {
  return this.setProductPackUnitUpserted(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.ProductsEvent.prototype.hasProductPackUnitUpserted = function() {
  return jspb.Message.getField(this, 7) != null;
};


/**
 * optional ProductPackUnitDeleted product_pack_unit_deleted = 8;
 * @return {?proto.bigblue.products.v0.ProductPackUnitDeleted}
 */
proto.bigblue.products.v0.ProductsEvent.prototype.getProductPackUnitDeleted = function() {
  return /** @type{?proto.bigblue.products.v0.ProductPackUnitDeleted} */ (
    jspb.Message.getWrapperField(this, proto.bigblue.products.v0.ProductPackUnitDeleted, 8));
};


/**
 * @param {?proto.bigblue.products.v0.ProductPackUnitDeleted|undefined} value
 * @return {!proto.bigblue.products.v0.ProductsEvent} returns this
*/
proto.bigblue.products.v0.ProductsEvent.prototype.setProductPackUnitDeleted = function(value) {
  return jspb.Message.setOneofWrapperField(this, 8, proto.bigblue.products.v0.ProductsEvent.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.ProductsEvent} returns this
 */
proto.bigblue.products.v0.ProductsEvent.prototype.clearProductPackUnitDeleted = function() {
  return this.setProductPackUnitDeleted(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.ProductsEvent.prototype.hasProductPackUnitDeleted = function() {
  return jspb.Message.getField(this, 8) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.ProductUpserted.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.ProductUpserted.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.ProductUpserted} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductUpserted.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, ""),
    correlationId: jspb.Message.getFieldWithDefault(msg, 2, ""),
    createTime: jspb.Message.getFieldWithDefault(msg, 3, ""),
    hash: jspb.Message.getFieldWithDefault(msg, 4, 0),
    organization: jspb.Message.getFieldWithDefault(msg, 5, ""),
    product: (f = msg.getProduct()) && bigblue_products_v0_product_pb.Product.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.ProductUpserted}
 */
proto.bigblue.products.v0.ProductUpserted.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.ProductUpserted;
  return proto.bigblue.products.v0.ProductUpserted.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.ProductUpserted} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.ProductUpserted}
 */
proto.bigblue.products.v0.ProductUpserted.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setCorrelationId(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setCreateTime(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readFixed64());
      msg.setHash(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setOrganization(value);
      break;
    case 6:
      var value = new bigblue_products_v0_product_pb.Product;
      reader.readMessage(value,bigblue_products_v0_product_pb.Product.deserializeBinaryFromReader);
      msg.setProduct(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.ProductUpserted.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.ProductUpserted.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.ProductUpserted} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductUpserted.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getCorrelationId();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getCreateTime();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getHash();
  if (f !== 0) {
    writer.writeFixed64(
      4,
      f
    );
  }
  f = message.getOrganization();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
  f = message.getProduct();
  if (f != null) {
    writer.writeMessage(
      6,
      f,
      bigblue_products_v0_product_pb.Product.serializeBinaryToWriter
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.bigblue.products.v0.ProductUpserted.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductUpserted} returns this
 */
proto.bigblue.products.v0.ProductUpserted.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string correlation_id = 2;
 * @return {string}
 */
proto.bigblue.products.v0.ProductUpserted.prototype.getCorrelationId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductUpserted} returns this
 */
proto.bigblue.products.v0.ProductUpserted.prototype.setCorrelationId = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string create_time = 3;
 * @return {string}
 */
proto.bigblue.products.v0.ProductUpserted.prototype.getCreateTime = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductUpserted} returns this
 */
proto.bigblue.products.v0.ProductUpserted.prototype.setCreateTime = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional fixed64 hash = 4;
 * @return {number}
 */
proto.bigblue.products.v0.ProductUpserted.prototype.getHash = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.bigblue.products.v0.ProductUpserted} returns this
 */
proto.bigblue.products.v0.ProductUpserted.prototype.setHash = function(value) {
  return jspb.Message.setProto3IntField(this, 4, value);
};


/**
 * optional string organization = 5;
 * @return {string}
 */
proto.bigblue.products.v0.ProductUpserted.prototype.getOrganization = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductUpserted} returns this
 */
proto.bigblue.products.v0.ProductUpserted.prototype.setOrganization = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};


/**
 * optional Product product = 6;
 * @return {?proto.bigblue.products.v0.Product}
 */
proto.bigblue.products.v0.ProductUpserted.prototype.getProduct = function() {
  return /** @type{?proto.bigblue.products.v0.Product} */ (
    jspb.Message.getWrapperField(this, bigblue_products_v0_product_pb.Product, 6));
};


/**
 * @param {?proto.bigblue.products.v0.Product|undefined} value
 * @return {!proto.bigblue.products.v0.ProductUpserted} returns this
*/
proto.bigblue.products.v0.ProductUpserted.prototype.setProduct = function(value) {
  return jspb.Message.setWrapperField(this, 6, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.ProductUpserted} returns this
 */
proto.bigblue.products.v0.ProductUpserted.prototype.clearProduct = function() {
  return this.setProduct(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.ProductUpserted.prototype.hasProduct = function() {
  return jspb.Message.getField(this, 6) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.ProductDeleted.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.ProductDeleted.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.ProductDeleted} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductDeleted.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, ""),
    correlationId: jspb.Message.getFieldWithDefault(msg, 2, ""),
    createTime: jspb.Message.getFieldWithDefault(msg, 3, ""),
    hash: jspb.Message.getFieldWithDefault(msg, 4, 0),
    organization: jspb.Message.getFieldWithDefault(msg, 5, ""),
    product: (f = msg.getProduct()) && bigblue_products_v0_product_pb.Product.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.ProductDeleted}
 */
proto.bigblue.products.v0.ProductDeleted.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.ProductDeleted;
  return proto.bigblue.products.v0.ProductDeleted.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.ProductDeleted} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.ProductDeleted}
 */
proto.bigblue.products.v0.ProductDeleted.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setCorrelationId(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setCreateTime(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readFixed64());
      msg.setHash(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setOrganization(value);
      break;
    case 6:
      var value = new bigblue_products_v0_product_pb.Product;
      reader.readMessage(value,bigblue_products_v0_product_pb.Product.deserializeBinaryFromReader);
      msg.setProduct(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.ProductDeleted.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.ProductDeleted.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.ProductDeleted} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductDeleted.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getCorrelationId();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getCreateTime();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getHash();
  if (f !== 0) {
    writer.writeFixed64(
      4,
      f
    );
  }
  f = message.getOrganization();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
  f = message.getProduct();
  if (f != null) {
    writer.writeMessage(
      6,
      f,
      bigblue_products_v0_product_pb.Product.serializeBinaryToWriter
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.bigblue.products.v0.ProductDeleted.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductDeleted} returns this
 */
proto.bigblue.products.v0.ProductDeleted.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string correlation_id = 2;
 * @return {string}
 */
proto.bigblue.products.v0.ProductDeleted.prototype.getCorrelationId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductDeleted} returns this
 */
proto.bigblue.products.v0.ProductDeleted.prototype.setCorrelationId = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string create_time = 3;
 * @return {string}
 */
proto.bigblue.products.v0.ProductDeleted.prototype.getCreateTime = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductDeleted} returns this
 */
proto.bigblue.products.v0.ProductDeleted.prototype.setCreateTime = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional fixed64 hash = 4;
 * @return {number}
 */
proto.bigblue.products.v0.ProductDeleted.prototype.getHash = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.bigblue.products.v0.ProductDeleted} returns this
 */
proto.bigblue.products.v0.ProductDeleted.prototype.setHash = function(value) {
  return jspb.Message.setProto3IntField(this, 4, value);
};


/**
 * optional string organization = 5;
 * @return {string}
 */
proto.bigblue.products.v0.ProductDeleted.prototype.getOrganization = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductDeleted} returns this
 */
proto.bigblue.products.v0.ProductDeleted.prototype.setOrganization = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};


/**
 * optional Product product = 6;
 * @return {?proto.bigblue.products.v0.Product}
 */
proto.bigblue.products.v0.ProductDeleted.prototype.getProduct = function() {
  return /** @type{?proto.bigblue.products.v0.Product} */ (
    jspb.Message.getWrapperField(this, bigblue_products_v0_product_pb.Product, 6));
};


/**
 * @param {?proto.bigblue.products.v0.Product|undefined} value
 * @return {!proto.bigblue.products.v0.ProductDeleted} returns this
*/
proto.bigblue.products.v0.ProductDeleted.prototype.setProduct = function(value) {
  return jspb.Message.setWrapperField(this, 6, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.ProductDeleted} returns this
 */
proto.bigblue.products.v0.ProductDeleted.prototype.clearProduct = function() {
  return this.setProduct(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.ProductDeleted.prototype.hasProduct = function() {
  return jspb.Message.getField(this, 6) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.BundleUpserted.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.BundleUpserted.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.BundleUpserted} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.BundleUpserted.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, ""),
    correlationId: jspb.Message.getFieldWithDefault(msg, 2, ""),
    createTime: jspb.Message.getFieldWithDefault(msg, 3, ""),
    hash: jspb.Message.getFieldWithDefault(msg, 4, 0),
    organization: jspb.Message.getFieldWithDefault(msg, 5, ""),
    bundle: (f = msg.getBundle()) && bigblue_products_v0_product_pb.Bundle.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.BundleUpserted}
 */
proto.bigblue.products.v0.BundleUpserted.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.BundleUpserted;
  return proto.bigblue.products.v0.BundleUpserted.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.BundleUpserted} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.BundleUpserted}
 */
proto.bigblue.products.v0.BundleUpserted.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setCorrelationId(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setCreateTime(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readFixed64());
      msg.setHash(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setOrganization(value);
      break;
    case 6:
      var value = new bigblue_products_v0_product_pb.Bundle;
      reader.readMessage(value,bigblue_products_v0_product_pb.Bundle.deserializeBinaryFromReader);
      msg.setBundle(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.BundleUpserted.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.BundleUpserted.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.BundleUpserted} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.BundleUpserted.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getCorrelationId();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getCreateTime();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getHash();
  if (f !== 0) {
    writer.writeFixed64(
      4,
      f
    );
  }
  f = message.getOrganization();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
  f = message.getBundle();
  if (f != null) {
    writer.writeMessage(
      6,
      f,
      bigblue_products_v0_product_pb.Bundle.serializeBinaryToWriter
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.bigblue.products.v0.BundleUpserted.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.BundleUpserted} returns this
 */
proto.bigblue.products.v0.BundleUpserted.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string correlation_id = 2;
 * @return {string}
 */
proto.bigblue.products.v0.BundleUpserted.prototype.getCorrelationId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.BundleUpserted} returns this
 */
proto.bigblue.products.v0.BundleUpserted.prototype.setCorrelationId = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string create_time = 3;
 * @return {string}
 */
proto.bigblue.products.v0.BundleUpserted.prototype.getCreateTime = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.BundleUpserted} returns this
 */
proto.bigblue.products.v0.BundleUpserted.prototype.setCreateTime = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional fixed64 hash = 4;
 * @return {number}
 */
proto.bigblue.products.v0.BundleUpserted.prototype.getHash = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.bigblue.products.v0.BundleUpserted} returns this
 */
proto.bigblue.products.v0.BundleUpserted.prototype.setHash = function(value) {
  return jspb.Message.setProto3IntField(this, 4, value);
};


/**
 * optional string organization = 5;
 * @return {string}
 */
proto.bigblue.products.v0.BundleUpserted.prototype.getOrganization = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.BundleUpserted} returns this
 */
proto.bigblue.products.v0.BundleUpserted.prototype.setOrganization = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};


/**
 * optional Bundle bundle = 6;
 * @return {?proto.bigblue.products.v0.Bundle}
 */
proto.bigblue.products.v0.BundleUpserted.prototype.getBundle = function() {
  return /** @type{?proto.bigblue.products.v0.Bundle} */ (
    jspb.Message.getWrapperField(this, bigblue_products_v0_product_pb.Bundle, 6));
};


/**
 * @param {?proto.bigblue.products.v0.Bundle|undefined} value
 * @return {!proto.bigblue.products.v0.BundleUpserted} returns this
*/
proto.bigblue.products.v0.BundleUpserted.prototype.setBundle = function(value) {
  return jspb.Message.setWrapperField(this, 6, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.BundleUpserted} returns this
 */
proto.bigblue.products.v0.BundleUpserted.prototype.clearBundle = function() {
  return this.setBundle(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.BundleUpserted.prototype.hasBundle = function() {
  return jspb.Message.getField(this, 6) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.BundleDeleted.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.BundleDeleted.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.BundleDeleted} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.BundleDeleted.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, ""),
    correlationId: jspb.Message.getFieldWithDefault(msg, 2, ""),
    createTime: jspb.Message.getFieldWithDefault(msg, 3, ""),
    hash: jspb.Message.getFieldWithDefault(msg, 4, 0),
    organization: jspb.Message.getFieldWithDefault(msg, 5, ""),
    bundle: (f = msg.getBundle()) && bigblue_products_v0_product_pb.Bundle.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.BundleDeleted}
 */
proto.bigblue.products.v0.BundleDeleted.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.BundleDeleted;
  return proto.bigblue.products.v0.BundleDeleted.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.BundleDeleted} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.BundleDeleted}
 */
proto.bigblue.products.v0.BundleDeleted.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setCorrelationId(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setCreateTime(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readFixed64());
      msg.setHash(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setOrganization(value);
      break;
    case 6:
      var value = new bigblue_products_v0_product_pb.Bundle;
      reader.readMessage(value,bigblue_products_v0_product_pb.Bundle.deserializeBinaryFromReader);
      msg.setBundle(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.BundleDeleted.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.BundleDeleted.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.BundleDeleted} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.BundleDeleted.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getCorrelationId();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getCreateTime();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getHash();
  if (f !== 0) {
    writer.writeFixed64(
      4,
      f
    );
  }
  f = message.getOrganization();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
  f = message.getBundle();
  if (f != null) {
    writer.writeMessage(
      6,
      f,
      bigblue_products_v0_product_pb.Bundle.serializeBinaryToWriter
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.bigblue.products.v0.BundleDeleted.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.BundleDeleted} returns this
 */
proto.bigblue.products.v0.BundleDeleted.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string correlation_id = 2;
 * @return {string}
 */
proto.bigblue.products.v0.BundleDeleted.prototype.getCorrelationId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.BundleDeleted} returns this
 */
proto.bigblue.products.v0.BundleDeleted.prototype.setCorrelationId = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string create_time = 3;
 * @return {string}
 */
proto.bigblue.products.v0.BundleDeleted.prototype.getCreateTime = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.BundleDeleted} returns this
 */
proto.bigblue.products.v0.BundleDeleted.prototype.setCreateTime = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional fixed64 hash = 4;
 * @return {number}
 */
proto.bigblue.products.v0.BundleDeleted.prototype.getHash = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.bigblue.products.v0.BundleDeleted} returns this
 */
proto.bigblue.products.v0.BundleDeleted.prototype.setHash = function(value) {
  return jspb.Message.setProto3IntField(this, 4, value);
};


/**
 * optional string organization = 5;
 * @return {string}
 */
proto.bigblue.products.v0.BundleDeleted.prototype.getOrganization = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.BundleDeleted} returns this
 */
proto.bigblue.products.v0.BundleDeleted.prototype.setOrganization = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};


/**
 * optional Bundle bundle = 6;
 * @return {?proto.bigblue.products.v0.Bundle}
 */
proto.bigblue.products.v0.BundleDeleted.prototype.getBundle = function() {
  return /** @type{?proto.bigblue.products.v0.Bundle} */ (
    jspb.Message.getWrapperField(this, bigblue_products_v0_product_pb.Bundle, 6));
};


/**
 * @param {?proto.bigblue.products.v0.Bundle|undefined} value
 * @return {!proto.bigblue.products.v0.BundleDeleted} returns this
*/
proto.bigblue.products.v0.BundleDeleted.prototype.setBundle = function(value) {
  return jspb.Message.setWrapperField(this, 6, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.BundleDeleted} returns this
 */
proto.bigblue.products.v0.BundleDeleted.prototype.clearBundle = function() {
  return this.setBundle(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.BundleDeleted.prototype.hasBundle = function() {
  return jspb.Message.getField(this, 6) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.ContainerUpserted.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.ContainerUpserted.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.ContainerUpserted} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ContainerUpserted.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, ""),
    correlationId: jspb.Message.getFieldWithDefault(msg, 2, ""),
    createTime: jspb.Message.getFieldWithDefault(msg, 3, ""),
    hash: jspb.Message.getFieldWithDefault(msg, 4, 0),
    organization: jspb.Message.getFieldWithDefault(msg, 5, ""),
    container: (f = msg.getContainer()) && bigblue_products_v0_container_pb.Container.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.ContainerUpserted}
 */
proto.bigblue.products.v0.ContainerUpserted.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.ContainerUpserted;
  return proto.bigblue.products.v0.ContainerUpserted.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.ContainerUpserted} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.ContainerUpserted}
 */
proto.bigblue.products.v0.ContainerUpserted.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setCorrelationId(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setCreateTime(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readFixed64());
      msg.setHash(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setOrganization(value);
      break;
    case 6:
      var value = new bigblue_products_v0_container_pb.Container;
      reader.readMessage(value,bigblue_products_v0_container_pb.Container.deserializeBinaryFromReader);
      msg.setContainer(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.ContainerUpserted.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.ContainerUpserted.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.ContainerUpserted} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ContainerUpserted.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getCorrelationId();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getCreateTime();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getHash();
  if (f !== 0) {
    writer.writeFixed64(
      4,
      f
    );
  }
  f = message.getOrganization();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
  f = message.getContainer();
  if (f != null) {
    writer.writeMessage(
      6,
      f,
      bigblue_products_v0_container_pb.Container.serializeBinaryToWriter
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.bigblue.products.v0.ContainerUpserted.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ContainerUpserted} returns this
 */
proto.bigblue.products.v0.ContainerUpserted.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string correlation_id = 2;
 * @return {string}
 */
proto.bigblue.products.v0.ContainerUpserted.prototype.getCorrelationId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ContainerUpserted} returns this
 */
proto.bigblue.products.v0.ContainerUpserted.prototype.setCorrelationId = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string create_time = 3;
 * @return {string}
 */
proto.bigblue.products.v0.ContainerUpserted.prototype.getCreateTime = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ContainerUpserted} returns this
 */
proto.bigblue.products.v0.ContainerUpserted.prototype.setCreateTime = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional fixed64 hash = 4;
 * @return {number}
 */
proto.bigblue.products.v0.ContainerUpserted.prototype.getHash = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.bigblue.products.v0.ContainerUpserted} returns this
 */
proto.bigblue.products.v0.ContainerUpserted.prototype.setHash = function(value) {
  return jspb.Message.setProto3IntField(this, 4, value);
};


/**
 * optional string organization = 5;
 * @return {string}
 */
proto.bigblue.products.v0.ContainerUpserted.prototype.getOrganization = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ContainerUpserted} returns this
 */
proto.bigblue.products.v0.ContainerUpserted.prototype.setOrganization = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};


/**
 * optional Container container = 6;
 * @return {?proto.bigblue.products.v0.Container}
 */
proto.bigblue.products.v0.ContainerUpserted.prototype.getContainer = function() {
  return /** @type{?proto.bigblue.products.v0.Container} */ (
    jspb.Message.getWrapperField(this, bigblue_products_v0_container_pb.Container, 6));
};


/**
 * @param {?proto.bigblue.products.v0.Container|undefined} value
 * @return {!proto.bigblue.products.v0.ContainerUpserted} returns this
*/
proto.bigblue.products.v0.ContainerUpserted.prototype.setContainer = function(value) {
  return jspb.Message.setWrapperField(this, 6, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.ContainerUpserted} returns this
 */
proto.bigblue.products.v0.ContainerUpserted.prototype.clearContainer = function() {
  return this.setContainer(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.ContainerUpserted.prototype.hasContainer = function() {
  return jspb.Message.getField(this, 6) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.ContainerDeleted.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.ContainerDeleted.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.ContainerDeleted} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ContainerDeleted.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, ""),
    correlationId: jspb.Message.getFieldWithDefault(msg, 2, ""),
    createTime: jspb.Message.getFieldWithDefault(msg, 3, ""),
    hash: jspb.Message.getFieldWithDefault(msg, 4, 0),
    organization: jspb.Message.getFieldWithDefault(msg, 5, ""),
    container: (f = msg.getContainer()) && bigblue_products_v0_container_pb.Container.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.ContainerDeleted}
 */
proto.bigblue.products.v0.ContainerDeleted.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.ContainerDeleted;
  return proto.bigblue.products.v0.ContainerDeleted.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.ContainerDeleted} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.ContainerDeleted}
 */
proto.bigblue.products.v0.ContainerDeleted.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setCorrelationId(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setCreateTime(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readFixed64());
      msg.setHash(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setOrganization(value);
      break;
    case 6:
      var value = new bigblue_products_v0_container_pb.Container;
      reader.readMessage(value,bigblue_products_v0_container_pb.Container.deserializeBinaryFromReader);
      msg.setContainer(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.ContainerDeleted.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.ContainerDeleted.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.ContainerDeleted} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ContainerDeleted.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getCorrelationId();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getCreateTime();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getHash();
  if (f !== 0) {
    writer.writeFixed64(
      4,
      f
    );
  }
  f = message.getOrganization();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
  f = message.getContainer();
  if (f != null) {
    writer.writeMessage(
      6,
      f,
      bigblue_products_v0_container_pb.Container.serializeBinaryToWriter
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.bigblue.products.v0.ContainerDeleted.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ContainerDeleted} returns this
 */
proto.bigblue.products.v0.ContainerDeleted.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string correlation_id = 2;
 * @return {string}
 */
proto.bigblue.products.v0.ContainerDeleted.prototype.getCorrelationId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ContainerDeleted} returns this
 */
proto.bigblue.products.v0.ContainerDeleted.prototype.setCorrelationId = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string create_time = 3;
 * @return {string}
 */
proto.bigblue.products.v0.ContainerDeleted.prototype.getCreateTime = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ContainerDeleted} returns this
 */
proto.bigblue.products.v0.ContainerDeleted.prototype.setCreateTime = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional fixed64 hash = 4;
 * @return {number}
 */
proto.bigblue.products.v0.ContainerDeleted.prototype.getHash = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.bigblue.products.v0.ContainerDeleted} returns this
 */
proto.bigblue.products.v0.ContainerDeleted.prototype.setHash = function(value) {
  return jspb.Message.setProto3IntField(this, 4, value);
};


/**
 * optional string organization = 5;
 * @return {string}
 */
proto.bigblue.products.v0.ContainerDeleted.prototype.getOrganization = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ContainerDeleted} returns this
 */
proto.bigblue.products.v0.ContainerDeleted.prototype.setOrganization = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};


/**
 * optional Container container = 6;
 * @return {?proto.bigblue.products.v0.Container}
 */
proto.bigblue.products.v0.ContainerDeleted.prototype.getContainer = function() {
  return /** @type{?proto.bigblue.products.v0.Container} */ (
    jspb.Message.getWrapperField(this, bigblue_products_v0_container_pb.Container, 6));
};


/**
 * @param {?proto.bigblue.products.v0.Container|undefined} value
 * @return {!proto.bigblue.products.v0.ContainerDeleted} returns this
*/
proto.bigblue.products.v0.ContainerDeleted.prototype.setContainer = function(value) {
  return jspb.Message.setWrapperField(this, 6, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.ContainerDeleted} returns this
 */
proto.bigblue.products.v0.ContainerDeleted.prototype.clearContainer = function() {
  return this.setContainer(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.ContainerDeleted.prototype.hasContainer = function() {
  return jspb.Message.getField(this, 6) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.ProductPackUnitUpserted.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.ProductPackUnitUpserted.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.ProductPackUnitUpserted} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductPackUnitUpserted.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, ""),
    correlationId: jspb.Message.getFieldWithDefault(msg, 2, ""),
    createTime: jspb.Message.getFieldWithDefault(msg, 3, ""),
    hash: jspb.Message.getFieldWithDefault(msg, 4, 0),
    organization: jspb.Message.getFieldWithDefault(msg, 5, ""),
    productPackUnit: (f = msg.getProductPackUnit()) && bigblue_products_v0_packunit_pb.ProductPackUnit.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.ProductPackUnitUpserted}
 */
proto.bigblue.products.v0.ProductPackUnitUpserted.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.ProductPackUnitUpserted;
  return proto.bigblue.products.v0.ProductPackUnitUpserted.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.ProductPackUnitUpserted} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.ProductPackUnitUpserted}
 */
proto.bigblue.products.v0.ProductPackUnitUpserted.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setCorrelationId(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setCreateTime(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readFixed64());
      msg.setHash(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setOrganization(value);
      break;
    case 6:
      var value = new bigblue_products_v0_packunit_pb.ProductPackUnit;
      reader.readMessage(value,bigblue_products_v0_packunit_pb.ProductPackUnit.deserializeBinaryFromReader);
      msg.setProductPackUnit(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.ProductPackUnitUpserted.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.ProductPackUnitUpserted.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.ProductPackUnitUpserted} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductPackUnitUpserted.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getCorrelationId();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getCreateTime();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getHash();
  if (f !== 0) {
    writer.writeFixed64(
      4,
      f
    );
  }
  f = message.getOrganization();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
  f = message.getProductPackUnit();
  if (f != null) {
    writer.writeMessage(
      6,
      f,
      bigblue_products_v0_packunit_pb.ProductPackUnit.serializeBinaryToWriter
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.bigblue.products.v0.ProductPackUnitUpserted.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductPackUnitUpserted} returns this
 */
proto.bigblue.products.v0.ProductPackUnitUpserted.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string correlation_id = 2;
 * @return {string}
 */
proto.bigblue.products.v0.ProductPackUnitUpserted.prototype.getCorrelationId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductPackUnitUpserted} returns this
 */
proto.bigblue.products.v0.ProductPackUnitUpserted.prototype.setCorrelationId = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string create_time = 3;
 * @return {string}
 */
proto.bigblue.products.v0.ProductPackUnitUpserted.prototype.getCreateTime = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductPackUnitUpserted} returns this
 */
proto.bigblue.products.v0.ProductPackUnitUpserted.prototype.setCreateTime = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional fixed64 hash = 4;
 * @return {number}
 */
proto.bigblue.products.v0.ProductPackUnitUpserted.prototype.getHash = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.bigblue.products.v0.ProductPackUnitUpserted} returns this
 */
proto.bigblue.products.v0.ProductPackUnitUpserted.prototype.setHash = function(value) {
  return jspb.Message.setProto3IntField(this, 4, value);
};


/**
 * optional string organization = 5;
 * @return {string}
 */
proto.bigblue.products.v0.ProductPackUnitUpserted.prototype.getOrganization = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductPackUnitUpserted} returns this
 */
proto.bigblue.products.v0.ProductPackUnitUpserted.prototype.setOrganization = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};


/**
 * optional ProductPackUnit product_pack_unit = 6;
 * @return {?proto.bigblue.products.v0.ProductPackUnit}
 */
proto.bigblue.products.v0.ProductPackUnitUpserted.prototype.getProductPackUnit = function() {
  return /** @type{?proto.bigblue.products.v0.ProductPackUnit} */ (
    jspb.Message.getWrapperField(this, bigblue_products_v0_packunit_pb.ProductPackUnit, 6));
};


/**
 * @param {?proto.bigblue.products.v0.ProductPackUnit|undefined} value
 * @return {!proto.bigblue.products.v0.ProductPackUnitUpserted} returns this
*/
proto.bigblue.products.v0.ProductPackUnitUpserted.prototype.setProductPackUnit = function(value) {
  return jspb.Message.setWrapperField(this, 6, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.ProductPackUnitUpserted} returns this
 */
proto.bigblue.products.v0.ProductPackUnitUpserted.prototype.clearProductPackUnit = function() {
  return this.setProductPackUnit(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.ProductPackUnitUpserted.prototype.hasProductPackUnit = function() {
  return jspb.Message.getField(this, 6) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.ProductPackUnitDeleted.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.ProductPackUnitDeleted.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.ProductPackUnitDeleted} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductPackUnitDeleted.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, ""),
    correlationId: jspb.Message.getFieldWithDefault(msg, 2, ""),
    createTime: jspb.Message.getFieldWithDefault(msg, 3, ""),
    hash: jspb.Message.getFieldWithDefault(msg, 4, 0),
    organization: jspb.Message.getFieldWithDefault(msg, 5, ""),
    productPackUnit: (f = msg.getProductPackUnit()) && bigblue_products_v0_packunit_pb.ProductPackUnit.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.ProductPackUnitDeleted}
 */
proto.bigblue.products.v0.ProductPackUnitDeleted.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.ProductPackUnitDeleted;
  return proto.bigblue.products.v0.ProductPackUnitDeleted.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.ProductPackUnitDeleted} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.ProductPackUnitDeleted}
 */
proto.bigblue.products.v0.ProductPackUnitDeleted.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setCorrelationId(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setCreateTime(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readFixed64());
      msg.setHash(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setOrganization(value);
      break;
    case 6:
      var value = new bigblue_products_v0_packunit_pb.ProductPackUnit;
      reader.readMessage(value,bigblue_products_v0_packunit_pb.ProductPackUnit.deserializeBinaryFromReader);
      msg.setProductPackUnit(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.ProductPackUnitDeleted.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.ProductPackUnitDeleted.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.ProductPackUnitDeleted} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductPackUnitDeleted.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getCorrelationId();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getCreateTime();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getHash();
  if (f !== 0) {
    writer.writeFixed64(
      4,
      f
    );
  }
  f = message.getOrganization();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
  f = message.getProductPackUnit();
  if (f != null) {
    writer.writeMessage(
      6,
      f,
      bigblue_products_v0_packunit_pb.ProductPackUnit.serializeBinaryToWriter
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.bigblue.products.v0.ProductPackUnitDeleted.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductPackUnitDeleted} returns this
 */
proto.bigblue.products.v0.ProductPackUnitDeleted.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string correlation_id = 2;
 * @return {string}
 */
proto.bigblue.products.v0.ProductPackUnitDeleted.prototype.getCorrelationId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductPackUnitDeleted} returns this
 */
proto.bigblue.products.v0.ProductPackUnitDeleted.prototype.setCorrelationId = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string create_time = 3;
 * @return {string}
 */
proto.bigblue.products.v0.ProductPackUnitDeleted.prototype.getCreateTime = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductPackUnitDeleted} returns this
 */
proto.bigblue.products.v0.ProductPackUnitDeleted.prototype.setCreateTime = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional fixed64 hash = 4;
 * @return {number}
 */
proto.bigblue.products.v0.ProductPackUnitDeleted.prototype.getHash = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.bigblue.products.v0.ProductPackUnitDeleted} returns this
 */
proto.bigblue.products.v0.ProductPackUnitDeleted.prototype.setHash = function(value) {
  return jspb.Message.setProto3IntField(this, 4, value);
};


/**
 * optional string organization = 5;
 * @return {string}
 */
proto.bigblue.products.v0.ProductPackUnitDeleted.prototype.getOrganization = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductPackUnitDeleted} returns this
 */
proto.bigblue.products.v0.ProductPackUnitDeleted.prototype.setOrganization = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};


/**
 * optional ProductPackUnit product_pack_unit = 6;
 * @return {?proto.bigblue.products.v0.ProductPackUnit}
 */
proto.bigblue.products.v0.ProductPackUnitDeleted.prototype.getProductPackUnit = function() {
  return /** @type{?proto.bigblue.products.v0.ProductPackUnit} */ (
    jspb.Message.getWrapperField(this, bigblue_products_v0_packunit_pb.ProductPackUnit, 6));
};


/**
 * @param {?proto.bigblue.products.v0.ProductPackUnit|undefined} value
 * @return {!proto.bigblue.products.v0.ProductPackUnitDeleted} returns this
*/
proto.bigblue.products.v0.ProductPackUnitDeleted.prototype.setProductPackUnit = function(value) {
  return jspb.Message.setWrapperField(this, 6, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.ProductPackUnitDeleted} returns this
 */
proto.bigblue.products.v0.ProductPackUnitDeleted.prototype.clearProductPackUnit = function() {
  return this.setProductPackUnit(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.ProductPackUnitDeleted.prototype.hasProductPackUnit = function() {
  return jspb.Message.getField(this, 6) != null;
};


goog.object.extend(exports, proto.bigblue.products.v0);
