// proto-ts-products/v0/product_pb.js
// source: bigblue/products/v0/product.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global =
    (typeof globalThis !== 'undefined' && globalThis) ||
    (typeof window !== 'undefined' && window) ||
    (typeof global !== 'undefined' && global) ||
    (typeof self !== 'undefined' && self) ||
    (function () { return this; }).call(null) ||
    Function('return this')();

var bigblue_common_event_pb = require('proto-ts-common/event_pb.js');
goog.object.extend(proto, bigblue_common_event_pb);
var bigblue_common_money_pb = require('proto-ts-common/money_pb.js');
goog.object.extend(proto, bigblue_common_money_pb);
goog.exportSymbol('proto.bigblue.products.v0.Bundle', null, global);
goog.exportSymbol('proto.bigblue.products.v0.BundleComponent', null, global);
goog.exportSymbol('proto.bigblue.products.v0.BundleVersion', null, global);
goog.exportSymbol('proto.bigblue.products.v0.HandlingType', null, global);
goog.exportSymbol('proto.bigblue.products.v0.Packaging', null, global);
goog.exportSymbol('proto.bigblue.products.v0.Product', null, global);
goog.exportSymbol('proto.bigblue.products.v0.ProductBuyerLocalization', null, global);
goog.exportSymbol('proto.bigblue.products.v0.ProductBuyerLocalizationOption', null, global);
goog.exportSymbol('proto.bigblue.products.v0.ProductBuyerLocalizationShopPriority', null, global);
goog.exportSymbol('proto.bigblue.products.v0.ProductForm', null, global);
goog.exportSymbol('proto.bigblue.products.v0.ProductImage', null, global);
goog.exportSymbol('proto.bigblue.products.v0.ProductNameTranslation', null, global);
goog.exportSymbol('proto.bigblue.products.v0.ProductOption', null, global);
goog.exportSymbol('proto.bigblue.products.v0.ProductVersion', null, global);
goog.exportSymbol('proto.bigblue.products.v0.Settings', null, global);
goog.exportSymbol('proto.bigblue.products.v0.StorageType', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.Settings = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.bigblue.products.v0.Settings, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.Settings.displayName = 'proto.bigblue.products.v0.Settings';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.Product = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.bigblue.products.v0.Product.repeatedFields_, null);
};
goog.inherits(proto.bigblue.products.v0.Product, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.Product.displayName = 'proto.bigblue.products.v0.Product';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.ProductForm = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.bigblue.products.v0.ProductForm.repeatedFields_, null);
};
goog.inherits(proto.bigblue.products.v0.ProductForm, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.ProductForm.displayName = 'proto.bigblue.products.v0.ProductForm';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.ProductOption = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.bigblue.products.v0.ProductOption, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.ProductOption.displayName = 'proto.bigblue.products.v0.ProductOption';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.ProductVersion = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.bigblue.products.v0.ProductVersion, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.ProductVersion.displayName = 'proto.bigblue.products.v0.ProductVersion';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.ProductNameTranslation = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.bigblue.products.v0.ProductNameTranslation, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.ProductNameTranslation.displayName = 'proto.bigblue.products.v0.ProductNameTranslation';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.Bundle = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.bigblue.products.v0.Bundle.repeatedFields_, null);
};
goog.inherits(proto.bigblue.products.v0.Bundle, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.Bundle.displayName = 'proto.bigblue.products.v0.Bundle';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.BundleComponent = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.bigblue.products.v0.BundleComponent, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.BundleComponent.displayName = 'proto.bigblue.products.v0.BundleComponent';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.BundleVersion = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.bigblue.products.v0.BundleVersion, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.BundleVersion.displayName = 'proto.bigblue.products.v0.BundleVersion';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.ProductImage = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.bigblue.products.v0.ProductImage, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.ProductImage.displayName = 'proto.bigblue.products.v0.ProductImage';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.ProductBuyerLocalizationOption = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.bigblue.products.v0.ProductBuyerLocalizationOption, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.ProductBuyerLocalizationOption.displayName = 'proto.bigblue.products.v0.ProductBuyerLocalizationOption';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.ProductBuyerLocalization = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.bigblue.products.v0.ProductBuyerLocalization.repeatedFields_, null);
};
goog.inherits(proto.bigblue.products.v0.ProductBuyerLocalization, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.ProductBuyerLocalization.displayName = 'proto.bigblue.products.v0.ProductBuyerLocalization';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.Settings.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.Settings.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.Settings} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.Settings.toObject = function(includeInstance, msg) {
  var f, obj = {
    tenant: jspb.Message.getFieldWithDefault(msg, 1, ""),
    lotsTracking: jspb.Message.getBooleanFieldWithDefault(msg, 2, false)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.Settings}
 */
proto.bigblue.products.v0.Settings.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.Settings;
  return proto.bigblue.products.v0.Settings.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.Settings} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.Settings}
 */
proto.bigblue.products.v0.Settings.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setTenant(value);
      break;
    case 2:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setLotsTracking(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.Settings.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.Settings.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.Settings} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.Settings.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getTenant();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getLotsTracking();
  if (f) {
    writer.writeBool(
      2,
      f
    );
  }
};


/**
 * optional string tenant = 1;
 * @return {string}
 */
proto.bigblue.products.v0.Settings.prototype.getTenant = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.Settings} returns this
 */
proto.bigblue.products.v0.Settings.prototype.setTenant = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional bool lots_tracking = 2;
 * @return {boolean}
 */
proto.bigblue.products.v0.Settings.prototype.getLotsTracking = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 2, false));
};


/**
 * @param {boolean} value
 * @return {!proto.bigblue.products.v0.Settings} returns this
 */
proto.bigblue.products.v0.Settings.prototype.setLotsTracking = function(value) {
  return jspb.Message.setProto3BooleanField(this, 2, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.bigblue.products.v0.Product.repeatedFields_ = [5,16];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.Product.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.Product.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.Product} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.Product.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, ""),
    tenant: jspb.Message.getFieldWithDefault(msg, 2, ""),
    displayName: jspb.Message.getFieldWithDefault(msg, 3, ""),
    description: jspb.Message.getFieldWithDefault(msg, 4, ""),
    optionsList: jspb.Message.toObjectList(msg.getOptionsList(),
    proto.bigblue.products.v0.ProductOption.toObject, includeInstance),
    barcode: jspb.Message.getFieldWithDefault(msg, 6, ""),
    weight: jspb.Message.getFieldWithDefault(msg, 7, 0),
    width: jspb.Message.getFieldWithDefault(msg, 8, 0),
    height: jspb.Message.getFieldWithDefault(msg, 9, 0),
    depth: jspb.Message.getFieldWithDefault(msg, 10, 0),
    packaging: jspb.Message.getFieldWithDefault(msg, 11, 0),
    value: (f = msg.getValue()) && bigblue_common_money_pb.Money.toObject(includeInstance, f),
    tariffNumber: jspb.Message.getFieldWithDefault(msg, 13, ""),
    originCountry: jspb.Message.getFieldWithDefault(msg, 14, ""),
    customsDescription: jspb.Message.getFieldWithDefault(msg, 15, ""),
    nameTranslationsList: jspb.Message.toObjectList(msg.getNameTranslationsList(),
    proto.bigblue.products.v0.ProductNameTranslation.toObject, includeInstance),
    foldable: jspb.Message.getBooleanFieldWithDefault(msg, 17, false),
    trackLots: jspb.Message.getBooleanFieldWithDefault(msg, 18, false),
    trackLotExpiry: jspb.Message.getBooleanFieldWithDefault(msg, 25, false),
    lotExpiryThresholdDays: jspb.Message.getFieldWithDefault(msg, 24, 0),
    marketingInsert: jspb.Message.getBooleanFieldWithDefault(msg, 19, false),
    shipWithPolybag: jspb.Message.getBooleanFieldWithDefault(msg, 20, false),
    storageType: jspb.Message.getFieldWithDefault(msg, 21, 0),
    handlingType: jspb.Message.getFieldWithDefault(msg, 22, 0),
    mainImage: (f = msg.getMainImage()) && proto.bigblue.products.v0.ProductImage.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.Product}
 */
proto.bigblue.products.v0.Product.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.Product;
  return proto.bigblue.products.v0.Product.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.Product} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.Product}
 */
proto.bigblue.products.v0.Product.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setTenant(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setDisplayName(value);
      break;
    case 4:
      var value = /** @type {string} */ (reader.readString());
      msg.setDescription(value);
      break;
    case 5:
      var value = new proto.bigblue.products.v0.ProductOption;
      reader.readMessage(value,proto.bigblue.products.v0.ProductOption.deserializeBinaryFromReader);
      msg.addOptions(value);
      break;
    case 6:
      var value = /** @type {string} */ (reader.readString());
      msg.setBarcode(value);
      break;
    case 7:
      var value = /** @type {number} */ (reader.readUint32());
      msg.setWeight(value);
      break;
    case 8:
      var value = /** @type {number} */ (reader.readUint32());
      msg.setWidth(value);
      break;
    case 9:
      var value = /** @type {number} */ (reader.readUint32());
      msg.setHeight(value);
      break;
    case 10:
      var value = /** @type {number} */ (reader.readUint32());
      msg.setDepth(value);
      break;
    case 11:
      var value = /** @type {!proto.bigblue.products.v0.Packaging} */ (reader.readEnum());
      msg.setPackaging(value);
      break;
    case 12:
      var value = new bigblue_common_money_pb.Money;
      reader.readMessage(value,bigblue_common_money_pb.Money.deserializeBinaryFromReader);
      msg.setValue(value);
      break;
    case 13:
      var value = /** @type {string} */ (reader.readString());
      msg.setTariffNumber(value);
      break;
    case 14:
      var value = /** @type {string} */ (reader.readString());
      msg.setOriginCountry(value);
      break;
    case 15:
      var value = /** @type {string} */ (reader.readString());
      msg.setCustomsDescription(value);
      break;
    case 16:
      var value = new proto.bigblue.products.v0.ProductNameTranslation;
      reader.readMessage(value,proto.bigblue.products.v0.ProductNameTranslation.deserializeBinaryFromReader);
      msg.addNameTranslations(value);
      break;
    case 17:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setFoldable(value);
      break;
    case 18:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setTrackLots(value);
      break;
    case 25:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setTrackLotExpiry(value);
      break;
    case 24:
      var value = /** @type {number} */ (reader.readUint32());
      msg.setLotExpiryThresholdDays(value);
      break;
    case 19:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setMarketingInsert(value);
      break;
    case 20:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setShipWithPolybag(value);
      break;
    case 21:
      var value = /** @type {!proto.bigblue.products.v0.StorageType} */ (reader.readEnum());
      msg.setStorageType(value);
      break;
    case 22:
      var value = /** @type {!proto.bigblue.products.v0.HandlingType} */ (reader.readEnum());
      msg.setHandlingType(value);
      break;
    case 23:
      var value = new proto.bigblue.products.v0.ProductImage;
      reader.readMessage(value,proto.bigblue.products.v0.ProductImage.deserializeBinaryFromReader);
      msg.setMainImage(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.Product.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.Product.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.Product} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.Product.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getTenant();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getDisplayName();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getDescription();
  if (f.length > 0) {
    writer.writeString(
      4,
      f
    );
  }
  f = message.getOptionsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      5,
      f,
      proto.bigblue.products.v0.ProductOption.serializeBinaryToWriter
    );
  }
  f = message.getBarcode();
  if (f.length > 0) {
    writer.writeString(
      6,
      f
    );
  }
  f = message.getWeight();
  if (f !== 0) {
    writer.writeUint32(
      7,
      f
    );
  }
  f = message.getWidth();
  if (f !== 0) {
    writer.writeUint32(
      8,
      f
    );
  }
  f = message.getHeight();
  if (f !== 0) {
    writer.writeUint32(
      9,
      f
    );
  }
  f = message.getDepth();
  if (f !== 0) {
    writer.writeUint32(
      10,
      f
    );
  }
  f = message.getPackaging();
  if (f !== 0.0) {
    writer.writeEnum(
      11,
      f
    );
  }
  f = message.getValue();
  if (f != null) {
    writer.writeMessage(
      12,
      f,
      bigblue_common_money_pb.Money.serializeBinaryToWriter
    );
  }
  f = message.getTariffNumber();
  if (f.length > 0) {
    writer.writeString(
      13,
      f
    );
  }
  f = message.getOriginCountry();
  if (f.length > 0) {
    writer.writeString(
      14,
      f
    );
  }
  f = message.getCustomsDescription();
  if (f.length > 0) {
    writer.writeString(
      15,
      f
    );
  }
  f = message.getNameTranslationsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      16,
      f,
      proto.bigblue.products.v0.ProductNameTranslation.serializeBinaryToWriter
    );
  }
  f = message.getFoldable();
  if (f) {
    writer.writeBool(
      17,
      f
    );
  }
  f = message.getTrackLots();
  if (f) {
    writer.writeBool(
      18,
      f
    );
  }
  f = message.getTrackLotExpiry();
  if (f) {
    writer.writeBool(
      25,
      f
    );
  }
  f = message.getLotExpiryThresholdDays();
  if (f !== 0) {
    writer.writeUint32(
      24,
      f
    );
  }
  f = message.getMarketingInsert();
  if (f) {
    writer.writeBool(
      19,
      f
    );
  }
  f = message.getShipWithPolybag();
  if (f) {
    writer.writeBool(
      20,
      f
    );
  }
  f = message.getStorageType();
  if (f !== 0.0) {
    writer.writeEnum(
      21,
      f
    );
  }
  f = message.getHandlingType();
  if (f !== 0.0) {
    writer.writeEnum(
      22,
      f
    );
  }
  f = message.getMainImage();
  if (f != null) {
    writer.writeMessage(
      23,
      f,
      proto.bigblue.products.v0.ProductImage.serializeBinaryToWriter
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.bigblue.products.v0.Product.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string tenant = 2;
 * @return {string}
 */
proto.bigblue.products.v0.Product.prototype.getTenant = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.setTenant = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string display_name = 3;
 * @return {string}
 */
proto.bigblue.products.v0.Product.prototype.getDisplayName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.setDisplayName = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional string description = 4;
 * @return {string}
 */
proto.bigblue.products.v0.Product.prototype.getDescription = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 4, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.setDescription = function(value) {
  return jspb.Message.setProto3StringField(this, 4, value);
};


/**
 * repeated ProductOption options = 5;
 * @return {!Array<!proto.bigblue.products.v0.ProductOption>}
 */
proto.bigblue.products.v0.Product.prototype.getOptionsList = function() {
  return /** @type{!Array<!proto.bigblue.products.v0.ProductOption>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.bigblue.products.v0.ProductOption, 5));
};


/**
 * @param {!Array<!proto.bigblue.products.v0.ProductOption>} value
 * @return {!proto.bigblue.products.v0.Product} returns this
*/
proto.bigblue.products.v0.Product.prototype.setOptionsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 5, value);
};


/**
 * @param {!proto.bigblue.products.v0.ProductOption=} opt_value
 * @param {number=} opt_index
 * @return {!proto.bigblue.products.v0.ProductOption}
 */
proto.bigblue.products.v0.Product.prototype.addOptions = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 5, opt_value, proto.bigblue.products.v0.ProductOption, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.clearOptionsList = function() {
  return this.setOptionsList([]);
};


/**
 * optional string barcode = 6;
 * @return {string}
 */
proto.bigblue.products.v0.Product.prototype.getBarcode = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 6, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.setBarcode = function(value) {
  return jspb.Message.setProto3StringField(this, 6, value);
};


/**
 * optional uint32 weight = 7;
 * @return {number}
 */
proto.bigblue.products.v0.Product.prototype.getWeight = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 7, 0));
};


/**
 * @param {number} value
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.setWeight = function(value) {
  return jspb.Message.setProto3IntField(this, 7, value);
};


/**
 * optional uint32 width = 8;
 * @return {number}
 */
proto.bigblue.products.v0.Product.prototype.getWidth = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 8, 0));
};


/**
 * @param {number} value
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.setWidth = function(value) {
  return jspb.Message.setProto3IntField(this, 8, value);
};


/**
 * optional uint32 height = 9;
 * @return {number}
 */
proto.bigblue.products.v0.Product.prototype.getHeight = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 9, 0));
};


/**
 * @param {number} value
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.setHeight = function(value) {
  return jspb.Message.setProto3IntField(this, 9, value);
};


/**
 * optional uint32 depth = 10;
 * @return {number}
 */
proto.bigblue.products.v0.Product.prototype.getDepth = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 10, 0));
};


/**
 * @param {number} value
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.setDepth = function(value) {
  return jspb.Message.setProto3IntField(this, 10, value);
};


/**
 * optional Packaging packaging = 11;
 * @return {!proto.bigblue.products.v0.Packaging}
 */
proto.bigblue.products.v0.Product.prototype.getPackaging = function() {
  return /** @type {!proto.bigblue.products.v0.Packaging} */ (jspb.Message.getFieldWithDefault(this, 11, 0));
};


/**
 * @param {!proto.bigblue.products.v0.Packaging} value
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.setPackaging = function(value) {
  return jspb.Message.setProto3EnumField(this, 11, value);
};


/**
 * optional bigblue.common.Money value = 12;
 * @return {?proto.bigblue.common.Money}
 */
proto.bigblue.products.v0.Product.prototype.getValue = function() {
  return /** @type{?proto.bigblue.common.Money} */ (
    jspb.Message.getWrapperField(this, bigblue_common_money_pb.Money, 12));
};


/**
 * @param {?proto.bigblue.common.Money|undefined} value
 * @return {!proto.bigblue.products.v0.Product} returns this
*/
proto.bigblue.products.v0.Product.prototype.setValue = function(value) {
  return jspb.Message.setWrapperField(this, 12, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.clearValue = function() {
  return this.setValue(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.Product.prototype.hasValue = function() {
  return jspb.Message.getField(this, 12) != null;
};


/**
 * optional string tariff_number = 13;
 * @return {string}
 */
proto.bigblue.products.v0.Product.prototype.getTariffNumber = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 13, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.setTariffNumber = function(value) {
  return jspb.Message.setProto3StringField(this, 13, value);
};


/**
 * optional string origin_country = 14;
 * @return {string}
 */
proto.bigblue.products.v0.Product.prototype.getOriginCountry = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 14, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.setOriginCountry = function(value) {
  return jspb.Message.setProto3StringField(this, 14, value);
};


/**
 * optional string customs_description = 15;
 * @return {string}
 */
proto.bigblue.products.v0.Product.prototype.getCustomsDescription = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 15, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.setCustomsDescription = function(value) {
  return jspb.Message.setProto3StringField(this, 15, value);
};


/**
 * repeated ProductNameTranslation name_translations = 16;
 * @return {!Array<!proto.bigblue.products.v0.ProductNameTranslation>}
 */
proto.bigblue.products.v0.Product.prototype.getNameTranslationsList = function() {
  return /** @type{!Array<!proto.bigblue.products.v0.ProductNameTranslation>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.bigblue.products.v0.ProductNameTranslation, 16));
};


/**
 * @param {!Array<!proto.bigblue.products.v0.ProductNameTranslation>} value
 * @return {!proto.bigblue.products.v0.Product} returns this
*/
proto.bigblue.products.v0.Product.prototype.setNameTranslationsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 16, value);
};


/**
 * @param {!proto.bigblue.products.v0.ProductNameTranslation=} opt_value
 * @param {number=} opt_index
 * @return {!proto.bigblue.products.v0.ProductNameTranslation}
 */
proto.bigblue.products.v0.Product.prototype.addNameTranslations = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 16, opt_value, proto.bigblue.products.v0.ProductNameTranslation, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.clearNameTranslationsList = function() {
  return this.setNameTranslationsList([]);
};


/**
 * optional bool foldable = 17;
 * @return {boolean}
 */
proto.bigblue.products.v0.Product.prototype.getFoldable = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 17, false));
};


/**
 * @param {boolean} value
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.setFoldable = function(value) {
  return jspb.Message.setProto3BooleanField(this, 17, value);
};


/**
 * optional bool track_lots = 18;
 * @return {boolean}
 */
proto.bigblue.products.v0.Product.prototype.getTrackLots = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 18, false));
};


/**
 * @param {boolean} value
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.setTrackLots = function(value) {
  return jspb.Message.setProto3BooleanField(this, 18, value);
};


/**
 * optional bool track_lot_expiry = 25;
 * @return {boolean}
 */
proto.bigblue.products.v0.Product.prototype.getTrackLotExpiry = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 25, false));
};


/**
 * @param {boolean} value
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.setTrackLotExpiry = function(value) {
  return jspb.Message.setProto3BooleanField(this, 25, value);
};


/**
 * optional uint32 lot_expiry_threshold_days = 24;
 * @return {number}
 */
proto.bigblue.products.v0.Product.prototype.getLotExpiryThresholdDays = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 24, 0));
};


/**
 * @param {number} value
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.setLotExpiryThresholdDays = function(value) {
  return jspb.Message.setProto3IntField(this, 24, value);
};


/**
 * optional bool marketing_insert = 19;
 * @return {boolean}
 */
proto.bigblue.products.v0.Product.prototype.getMarketingInsert = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 19, false));
};


/**
 * @param {boolean} value
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.setMarketingInsert = function(value) {
  return jspb.Message.setProto3BooleanField(this, 19, value);
};


/**
 * optional bool ship_with_polybag = 20;
 * @return {boolean}
 */
proto.bigblue.products.v0.Product.prototype.getShipWithPolybag = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 20, false));
};


/**
 * @param {boolean} value
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.setShipWithPolybag = function(value) {
  return jspb.Message.setProto3BooleanField(this, 20, value);
};


/**
 * optional StorageType storage_type = 21;
 * @return {!proto.bigblue.products.v0.StorageType}
 */
proto.bigblue.products.v0.Product.prototype.getStorageType = function() {
  return /** @type {!proto.bigblue.products.v0.StorageType} */ (jspb.Message.getFieldWithDefault(this, 21, 0));
};


/**
 * @param {!proto.bigblue.products.v0.StorageType} value
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.setStorageType = function(value) {
  return jspb.Message.setProto3EnumField(this, 21, value);
};


/**
 * optional HandlingType handling_type = 22;
 * @return {!proto.bigblue.products.v0.HandlingType}
 */
proto.bigblue.products.v0.Product.prototype.getHandlingType = function() {
  return /** @type {!proto.bigblue.products.v0.HandlingType} */ (jspb.Message.getFieldWithDefault(this, 22, 0));
};


/**
 * @param {!proto.bigblue.products.v0.HandlingType} value
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.setHandlingType = function(value) {
  return jspb.Message.setProto3EnumField(this, 22, value);
};


/**
 * optional ProductImage main_image = 23;
 * @return {?proto.bigblue.products.v0.ProductImage}
 */
proto.bigblue.products.v0.Product.prototype.getMainImage = function() {
  return /** @type{?proto.bigblue.products.v0.ProductImage} */ (
    jspb.Message.getWrapperField(this, proto.bigblue.products.v0.ProductImage, 23));
};


/**
 * @param {?proto.bigblue.products.v0.ProductImage|undefined} value
 * @return {!proto.bigblue.products.v0.Product} returns this
*/
proto.bigblue.products.v0.Product.prototype.setMainImage = function(value) {
  return jspb.Message.setWrapperField(this, 23, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.Product} returns this
 */
proto.bigblue.products.v0.Product.prototype.clearMainImage = function() {
  return this.setMainImage(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.Product.prototype.hasMainImage = function() {
  return jspb.Message.getField(this, 23) != null;
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.bigblue.products.v0.ProductForm.repeatedFields_ = [5,16];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.ProductForm.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.ProductForm.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.ProductForm} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductForm.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, ""),
    displayName: jspb.Message.getFieldWithDefault(msg, 3, ""),
    description: jspb.Message.getFieldWithDefault(msg, 4, ""),
    optionsList: jspb.Message.toObjectList(msg.getOptionsList(),
    proto.bigblue.products.v0.ProductOption.toObject, includeInstance),
    barcode: jspb.Message.getFieldWithDefault(msg, 6, ""),
    weight: jspb.Message.getFieldWithDefault(msg, 7, 0),
    width: jspb.Message.getFieldWithDefault(msg, 8, 0),
    height: jspb.Message.getFieldWithDefault(msg, 9, 0),
    depth: jspb.Message.getFieldWithDefault(msg, 10, 0),
    packaging: jspb.Message.getFieldWithDefault(msg, 11, 0),
    value: (f = msg.getValue()) && bigblue_common_money_pb.Money.toObject(includeInstance, f),
    tariffNumber: jspb.Message.getFieldWithDefault(msg, 13, ""),
    originCountry: jspb.Message.getFieldWithDefault(msg, 14, ""),
    customsDescription: jspb.Message.getFieldWithDefault(msg, 15, ""),
    nameTranslationsList: jspb.Message.toObjectList(msg.getNameTranslationsList(),
    proto.bigblue.products.v0.ProductNameTranslation.toObject, includeInstance),
    foldable: jspb.Message.getBooleanFieldWithDefault(msg, 17, false),
    trackLots: jspb.Message.getBooleanFieldWithDefault(msg, 18, false),
    trackLotExpiry: jspb.Message.getBooleanFieldWithDefault(msg, 25, false),
    lotExpiryThresholdDays: jspb.Message.getFieldWithDefault(msg, 24, 0),
    marketingInsert: jspb.Message.getBooleanFieldWithDefault(msg, 19, false),
    shipWithPolybag: jspb.Message.getBooleanFieldWithDefault(msg, 20, false),
    storageType: jspb.Message.getFieldWithDefault(msg, 21, 0),
    handlingType: jspb.Message.getFieldWithDefault(msg, 22, 0),
    mainImage: (f = msg.getMainImage()) && proto.bigblue.products.v0.ProductImage.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.ProductForm}
 */
proto.bigblue.products.v0.ProductForm.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.ProductForm;
  return proto.bigblue.products.v0.ProductForm.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.ProductForm} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.ProductForm}
 */
proto.bigblue.products.v0.ProductForm.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setDisplayName(value);
      break;
    case 4:
      var value = /** @type {string} */ (reader.readString());
      msg.setDescription(value);
      break;
    case 5:
      var value = new proto.bigblue.products.v0.ProductOption;
      reader.readMessage(value,proto.bigblue.products.v0.ProductOption.deserializeBinaryFromReader);
      msg.addOptions(value);
      break;
    case 6:
      var value = /** @type {string} */ (reader.readString());
      msg.setBarcode(value);
      break;
    case 7:
      var value = /** @type {number} */ (reader.readUint32());
      msg.setWeight(value);
      break;
    case 8:
      var value = /** @type {number} */ (reader.readUint32());
      msg.setWidth(value);
      break;
    case 9:
      var value = /** @type {number} */ (reader.readUint32());
      msg.setHeight(value);
      break;
    case 10:
      var value = /** @type {number} */ (reader.readUint32());
      msg.setDepth(value);
      break;
    case 11:
      var value = /** @type {!proto.bigblue.products.v0.Packaging} */ (reader.readEnum());
      msg.setPackaging(value);
      break;
    case 12:
      var value = new bigblue_common_money_pb.Money;
      reader.readMessage(value,bigblue_common_money_pb.Money.deserializeBinaryFromReader);
      msg.setValue(value);
      break;
    case 13:
      var value = /** @type {string} */ (reader.readString());
      msg.setTariffNumber(value);
      break;
    case 14:
      var value = /** @type {string} */ (reader.readString());
      msg.setOriginCountry(value);
      break;
    case 15:
      var value = /** @type {string} */ (reader.readString());
      msg.setCustomsDescription(value);
      break;
    case 16:
      var value = new proto.bigblue.products.v0.ProductNameTranslation;
      reader.readMessage(value,proto.bigblue.products.v0.ProductNameTranslation.deserializeBinaryFromReader);
      msg.addNameTranslations(value);
      break;
    case 17:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setFoldable(value);
      break;
    case 18:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setTrackLots(value);
      break;
    case 25:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setTrackLotExpiry(value);
      break;
    case 24:
      var value = /** @type {number} */ (reader.readUint32());
      msg.setLotExpiryThresholdDays(value);
      break;
    case 19:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setMarketingInsert(value);
      break;
    case 20:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setShipWithPolybag(value);
      break;
    case 21:
      var value = /** @type {!proto.bigblue.products.v0.StorageType} */ (reader.readEnum());
      msg.setStorageType(value);
      break;
    case 22:
      var value = /** @type {!proto.bigblue.products.v0.HandlingType} */ (reader.readEnum());
      msg.setHandlingType(value);
      break;
    case 23:
      var value = new proto.bigblue.products.v0.ProductImage;
      reader.readMessage(value,proto.bigblue.products.v0.ProductImage.deserializeBinaryFromReader);
      msg.setMainImage(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.ProductForm.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.ProductForm.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.ProductForm} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductForm.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getDisplayName();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getDescription();
  if (f.length > 0) {
    writer.writeString(
      4,
      f
    );
  }
  f = message.getOptionsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      5,
      f,
      proto.bigblue.products.v0.ProductOption.serializeBinaryToWriter
    );
  }
  f = message.getBarcode();
  if (f.length > 0) {
    writer.writeString(
      6,
      f
    );
  }
  f = message.getWeight();
  if (f !== 0) {
    writer.writeUint32(
      7,
      f
    );
  }
  f = message.getWidth();
  if (f !== 0) {
    writer.writeUint32(
      8,
      f
    );
  }
  f = message.getHeight();
  if (f !== 0) {
    writer.writeUint32(
      9,
      f
    );
  }
  f = message.getDepth();
  if (f !== 0) {
    writer.writeUint32(
      10,
      f
    );
  }
  f = message.getPackaging();
  if (f !== 0.0) {
    writer.writeEnum(
      11,
      f
    );
  }
  f = message.getValue();
  if (f != null) {
    writer.writeMessage(
      12,
      f,
      bigblue_common_money_pb.Money.serializeBinaryToWriter
    );
  }
  f = message.getTariffNumber();
  if (f.length > 0) {
    writer.writeString(
      13,
      f
    );
  }
  f = message.getOriginCountry();
  if (f.length > 0) {
    writer.writeString(
      14,
      f
    );
  }
  f = message.getCustomsDescription();
  if (f.length > 0) {
    writer.writeString(
      15,
      f
    );
  }
  f = message.getNameTranslationsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      16,
      f,
      proto.bigblue.products.v0.ProductNameTranslation.serializeBinaryToWriter
    );
  }
  f = message.getFoldable();
  if (f) {
    writer.writeBool(
      17,
      f
    );
  }
  f = message.getTrackLots();
  if (f) {
    writer.writeBool(
      18,
      f
    );
  }
  f = message.getTrackLotExpiry();
  if (f) {
    writer.writeBool(
      25,
      f
    );
  }
  f = message.getLotExpiryThresholdDays();
  if (f !== 0) {
    writer.writeUint32(
      24,
      f
    );
  }
  f = message.getMarketingInsert();
  if (f) {
    writer.writeBool(
      19,
      f
    );
  }
  f = message.getShipWithPolybag();
  if (f) {
    writer.writeBool(
      20,
      f
    );
  }
  f = message.getStorageType();
  if (f !== 0.0) {
    writer.writeEnum(
      21,
      f
    );
  }
  f = message.getHandlingType();
  if (f !== 0.0) {
    writer.writeEnum(
      22,
      f
    );
  }
  f = message.getMainImage();
  if (f != null) {
    writer.writeMessage(
      23,
      f,
      proto.bigblue.products.v0.ProductImage.serializeBinaryToWriter
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.bigblue.products.v0.ProductForm.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string display_name = 3;
 * @return {string}
 */
proto.bigblue.products.v0.ProductForm.prototype.getDisplayName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.setDisplayName = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional string description = 4;
 * @return {string}
 */
proto.bigblue.products.v0.ProductForm.prototype.getDescription = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 4, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.setDescription = function(value) {
  return jspb.Message.setProto3StringField(this, 4, value);
};


/**
 * repeated ProductOption options = 5;
 * @return {!Array<!proto.bigblue.products.v0.ProductOption>}
 */
proto.bigblue.products.v0.ProductForm.prototype.getOptionsList = function() {
  return /** @type{!Array<!proto.bigblue.products.v0.ProductOption>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.bigblue.products.v0.ProductOption, 5));
};


/**
 * @param {!Array<!proto.bigblue.products.v0.ProductOption>} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
*/
proto.bigblue.products.v0.ProductForm.prototype.setOptionsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 5, value);
};


/**
 * @param {!proto.bigblue.products.v0.ProductOption=} opt_value
 * @param {number=} opt_index
 * @return {!proto.bigblue.products.v0.ProductOption}
 */
proto.bigblue.products.v0.ProductForm.prototype.addOptions = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 5, opt_value, proto.bigblue.products.v0.ProductOption, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.clearOptionsList = function() {
  return this.setOptionsList([]);
};


/**
 * optional string barcode = 6;
 * @return {string}
 */
proto.bigblue.products.v0.ProductForm.prototype.getBarcode = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 6, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.setBarcode = function(value) {
  return jspb.Message.setProto3StringField(this, 6, value);
};


/**
 * optional uint32 weight = 7;
 * @return {number}
 */
proto.bigblue.products.v0.ProductForm.prototype.getWeight = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 7, 0));
};


/**
 * @param {number} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.setWeight = function(value) {
  return jspb.Message.setProto3IntField(this, 7, value);
};


/**
 * optional uint32 width = 8;
 * @return {number}
 */
proto.bigblue.products.v0.ProductForm.prototype.getWidth = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 8, 0));
};


/**
 * @param {number} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.setWidth = function(value) {
  return jspb.Message.setProto3IntField(this, 8, value);
};


/**
 * optional uint32 height = 9;
 * @return {number}
 */
proto.bigblue.products.v0.ProductForm.prototype.getHeight = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 9, 0));
};


/**
 * @param {number} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.setHeight = function(value) {
  return jspb.Message.setProto3IntField(this, 9, value);
};


/**
 * optional uint32 depth = 10;
 * @return {number}
 */
proto.bigblue.products.v0.ProductForm.prototype.getDepth = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 10, 0));
};


/**
 * @param {number} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.setDepth = function(value) {
  return jspb.Message.setProto3IntField(this, 10, value);
};


/**
 * optional Packaging packaging = 11;
 * @return {!proto.bigblue.products.v0.Packaging}
 */
proto.bigblue.products.v0.ProductForm.prototype.getPackaging = function() {
  return /** @type {!proto.bigblue.products.v0.Packaging} */ (jspb.Message.getFieldWithDefault(this, 11, 0));
};


/**
 * @param {!proto.bigblue.products.v0.Packaging} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.setPackaging = function(value) {
  return jspb.Message.setProto3EnumField(this, 11, value);
};


/**
 * optional bigblue.common.Money value = 12;
 * @return {?proto.bigblue.common.Money}
 */
proto.bigblue.products.v0.ProductForm.prototype.getValue = function() {
  return /** @type{?proto.bigblue.common.Money} */ (
    jspb.Message.getWrapperField(this, bigblue_common_money_pb.Money, 12));
};


/**
 * @param {?proto.bigblue.common.Money|undefined} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
*/
proto.bigblue.products.v0.ProductForm.prototype.setValue = function(value) {
  return jspb.Message.setWrapperField(this, 12, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.clearValue = function() {
  return this.setValue(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.ProductForm.prototype.hasValue = function() {
  return jspb.Message.getField(this, 12) != null;
};


/**
 * optional string tariff_number = 13;
 * @return {string}
 */
proto.bigblue.products.v0.ProductForm.prototype.getTariffNumber = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 13, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.setTariffNumber = function(value) {
  return jspb.Message.setProto3StringField(this, 13, value);
};


/**
 * optional string origin_country = 14;
 * @return {string}
 */
proto.bigblue.products.v0.ProductForm.prototype.getOriginCountry = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 14, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.setOriginCountry = function(value) {
  return jspb.Message.setProto3StringField(this, 14, value);
};


/**
 * optional string customs_description = 15;
 * @return {string}
 */
proto.bigblue.products.v0.ProductForm.prototype.getCustomsDescription = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 15, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.setCustomsDescription = function(value) {
  return jspb.Message.setProto3StringField(this, 15, value);
};


/**
 * repeated ProductNameTranslation name_translations = 16;
 * @return {!Array<!proto.bigblue.products.v0.ProductNameTranslation>}
 */
proto.bigblue.products.v0.ProductForm.prototype.getNameTranslationsList = function() {
  return /** @type{!Array<!proto.bigblue.products.v0.ProductNameTranslation>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.bigblue.products.v0.ProductNameTranslation, 16));
};


/**
 * @param {!Array<!proto.bigblue.products.v0.ProductNameTranslation>} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
*/
proto.bigblue.products.v0.ProductForm.prototype.setNameTranslationsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 16, value);
};


/**
 * @param {!proto.bigblue.products.v0.ProductNameTranslation=} opt_value
 * @param {number=} opt_index
 * @return {!proto.bigblue.products.v0.ProductNameTranslation}
 */
proto.bigblue.products.v0.ProductForm.prototype.addNameTranslations = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 16, opt_value, proto.bigblue.products.v0.ProductNameTranslation, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.clearNameTranslationsList = function() {
  return this.setNameTranslationsList([]);
};


/**
 * optional bool foldable = 17;
 * @return {boolean}
 */
proto.bigblue.products.v0.ProductForm.prototype.getFoldable = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 17, false));
};


/**
 * @param {boolean} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.setFoldable = function(value) {
  return jspb.Message.setProto3BooleanField(this, 17, value);
};


/**
 * optional bool track_lots = 18;
 * @return {boolean}
 */
proto.bigblue.products.v0.ProductForm.prototype.getTrackLots = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 18, false));
};


/**
 * @param {boolean} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.setTrackLots = function(value) {
  return jspb.Message.setProto3BooleanField(this, 18, value);
};


/**
 * optional bool track_lot_expiry = 25;
 * @return {boolean}
 */
proto.bigblue.products.v0.ProductForm.prototype.getTrackLotExpiry = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 25, false));
};


/**
 * @param {boolean} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.setTrackLotExpiry = function(value) {
  return jspb.Message.setProto3BooleanField(this, 25, value);
};


/**
 * optional uint32 lot_expiry_threshold_days = 24;
 * @return {number}
 */
proto.bigblue.products.v0.ProductForm.prototype.getLotExpiryThresholdDays = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 24, 0));
};


/**
 * @param {number} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.setLotExpiryThresholdDays = function(value) {
  return jspb.Message.setProto3IntField(this, 24, value);
};


/**
 * optional bool marketing_insert = 19;
 * @return {boolean}
 */
proto.bigblue.products.v0.ProductForm.prototype.getMarketingInsert = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 19, false));
};


/**
 * @param {boolean} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.setMarketingInsert = function(value) {
  return jspb.Message.setProto3BooleanField(this, 19, value);
};


/**
 * optional bool ship_with_polybag = 20;
 * @return {boolean}
 */
proto.bigblue.products.v0.ProductForm.prototype.getShipWithPolybag = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 20, false));
};


/**
 * @param {boolean} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.setShipWithPolybag = function(value) {
  return jspb.Message.setProto3BooleanField(this, 20, value);
};


/**
 * optional StorageType storage_type = 21;
 * @return {!proto.bigblue.products.v0.StorageType}
 */
proto.bigblue.products.v0.ProductForm.prototype.getStorageType = function() {
  return /** @type {!proto.bigblue.products.v0.StorageType} */ (jspb.Message.getFieldWithDefault(this, 21, 0));
};


/**
 * @param {!proto.bigblue.products.v0.StorageType} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.setStorageType = function(value) {
  return jspb.Message.setProto3EnumField(this, 21, value);
};


/**
 * optional HandlingType handling_type = 22;
 * @return {!proto.bigblue.products.v0.HandlingType}
 */
proto.bigblue.products.v0.ProductForm.prototype.getHandlingType = function() {
  return /** @type {!proto.bigblue.products.v0.HandlingType} */ (jspb.Message.getFieldWithDefault(this, 22, 0));
};


/**
 * @param {!proto.bigblue.products.v0.HandlingType} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.setHandlingType = function(value) {
  return jspb.Message.setProto3EnumField(this, 22, value);
};


/**
 * optional ProductImage main_image = 23;
 * @return {?proto.bigblue.products.v0.ProductImage}
 */
proto.bigblue.products.v0.ProductForm.prototype.getMainImage = function() {
  return /** @type{?proto.bigblue.products.v0.ProductImage} */ (
    jspb.Message.getWrapperField(this, proto.bigblue.products.v0.ProductImage, 23));
};


/**
 * @param {?proto.bigblue.products.v0.ProductImage|undefined} value
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
*/
proto.bigblue.products.v0.ProductForm.prototype.setMainImage = function(value) {
  return jspb.Message.setWrapperField(this, 23, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.ProductForm} returns this
 */
proto.bigblue.products.v0.ProductForm.prototype.clearMainImage = function() {
  return this.setMainImage(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.ProductForm.prototype.hasMainImage = function() {
  return jspb.Message.getField(this, 23) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.ProductOption.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.ProductOption.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.ProductOption} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductOption.toObject = function(includeInstance, msg) {
  var f, obj = {
    name: jspb.Message.getFieldWithDefault(msg, 1, ""),
    value: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.ProductOption}
 */
proto.bigblue.products.v0.ProductOption.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.ProductOption;
  return proto.bigblue.products.v0.ProductOption.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.ProductOption} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.ProductOption}
 */
proto.bigblue.products.v0.ProductOption.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setName(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setValue(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.ProductOption.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.ProductOption.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.ProductOption} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductOption.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getName();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getValue();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional string name = 1;
 * @return {string}
 */
proto.bigblue.products.v0.ProductOption.prototype.getName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductOption} returns this
 */
proto.bigblue.products.v0.ProductOption.prototype.setName = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string value = 2;
 * @return {string}
 */
proto.bigblue.products.v0.ProductOption.prototype.getValue = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductOption} returns this
 */
proto.bigblue.products.v0.ProductOption.prototype.setValue = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.ProductVersion.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.ProductVersion.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.ProductVersion} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductVersion.toObject = function(includeInstance, msg) {
  var f, obj = {
    correlationId: jspb.Message.getFieldWithDefault(msg, 1, ""),
    createTime: jspb.Message.getFieldWithDefault(msg, 2, ""),
    hash: jspb.Message.getFieldWithDefault(msg, 3, 0),
    changeType: jspb.Message.getFieldWithDefault(msg, 4, 0),
    product: (f = msg.getProduct()) && proto.bigblue.products.v0.Product.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.ProductVersion}
 */
proto.bigblue.products.v0.ProductVersion.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.ProductVersion;
  return proto.bigblue.products.v0.ProductVersion.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.ProductVersion} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.ProductVersion}
 */
proto.bigblue.products.v0.ProductVersion.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setCorrelationId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setCreateTime(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readFixed64());
      msg.setHash(value);
      break;
    case 4:
      var value = /** @type {!proto.bigblue.common.ChangeType} */ (reader.readEnum());
      msg.setChangeType(value);
      break;
    case 5:
      var value = new proto.bigblue.products.v0.Product;
      reader.readMessage(value,proto.bigblue.products.v0.Product.deserializeBinaryFromReader);
      msg.setProduct(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.ProductVersion.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.ProductVersion.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.ProductVersion} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductVersion.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getCorrelationId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getCreateTime();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getHash();
  if (f !== 0) {
    writer.writeFixed64(
      3,
      f
    );
  }
  f = message.getChangeType();
  if (f !== 0.0) {
    writer.writeEnum(
      4,
      f
    );
  }
  f = message.getProduct();
  if (f != null) {
    writer.writeMessage(
      5,
      f,
      proto.bigblue.products.v0.Product.serializeBinaryToWriter
    );
  }
};


/**
 * optional string correlation_id = 1;
 * @return {string}
 */
proto.bigblue.products.v0.ProductVersion.prototype.getCorrelationId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductVersion} returns this
 */
proto.bigblue.products.v0.ProductVersion.prototype.setCorrelationId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string create_time = 2;
 * @return {string}
 */
proto.bigblue.products.v0.ProductVersion.prototype.getCreateTime = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductVersion} returns this
 */
proto.bigblue.products.v0.ProductVersion.prototype.setCreateTime = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional fixed64 hash = 3;
 * @return {number}
 */
proto.bigblue.products.v0.ProductVersion.prototype.getHash = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {number} value
 * @return {!proto.bigblue.products.v0.ProductVersion} returns this
 */
proto.bigblue.products.v0.ProductVersion.prototype.setHash = function(value) {
  return jspb.Message.setProto3IntField(this, 3, value);
};


/**
 * optional bigblue.common.ChangeType change_type = 4;
 * @return {!proto.bigblue.common.ChangeType}
 */
proto.bigblue.products.v0.ProductVersion.prototype.getChangeType = function() {
  return /** @type {!proto.bigblue.common.ChangeType} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {!proto.bigblue.common.ChangeType} value
 * @return {!proto.bigblue.products.v0.ProductVersion} returns this
 */
proto.bigblue.products.v0.ProductVersion.prototype.setChangeType = function(value) {
  return jspb.Message.setProto3EnumField(this, 4, value);
};


/**
 * optional Product product = 5;
 * @return {?proto.bigblue.products.v0.Product}
 */
proto.bigblue.products.v0.ProductVersion.prototype.getProduct = function() {
  return /** @type{?proto.bigblue.products.v0.Product} */ (
    jspb.Message.getWrapperField(this, proto.bigblue.products.v0.Product, 5));
};


/**
 * @param {?proto.bigblue.products.v0.Product|undefined} value
 * @return {!proto.bigblue.products.v0.ProductVersion} returns this
*/
proto.bigblue.products.v0.ProductVersion.prototype.setProduct = function(value) {
  return jspb.Message.setWrapperField(this, 5, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.ProductVersion} returns this
 */
proto.bigblue.products.v0.ProductVersion.prototype.clearProduct = function() {
  return this.setProduct(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.ProductVersion.prototype.hasProduct = function() {
  return jspb.Message.getField(this, 5) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.ProductNameTranslation.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.ProductNameTranslation.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.ProductNameTranslation} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductNameTranslation.toObject = function(includeInstance, msg) {
  var f, obj = {
    language: jspb.Message.getFieldWithDefault(msg, 1, ""),
    name: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.ProductNameTranslation}
 */
proto.bigblue.products.v0.ProductNameTranslation.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.ProductNameTranslation;
  return proto.bigblue.products.v0.ProductNameTranslation.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.ProductNameTranslation} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.ProductNameTranslation}
 */
proto.bigblue.products.v0.ProductNameTranslation.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setLanguage(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setName(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.ProductNameTranslation.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.ProductNameTranslation.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.ProductNameTranslation} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductNameTranslation.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getLanguage();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getName();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional string language = 1;
 * @return {string}
 */
proto.bigblue.products.v0.ProductNameTranslation.prototype.getLanguage = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductNameTranslation} returns this
 */
proto.bigblue.products.v0.ProductNameTranslation.prototype.setLanguage = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string name = 2;
 * @return {string}
 */
proto.bigblue.products.v0.ProductNameTranslation.prototype.getName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductNameTranslation} returns this
 */
proto.bigblue.products.v0.ProductNameTranslation.prototype.setName = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.bigblue.products.v0.Bundle.repeatedFields_ = [3,4];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.Bundle.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.Bundle.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.Bundle} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.Bundle.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, ""),
    displayName: jspb.Message.getFieldWithDefault(msg, 2, ""),
    componentsList: jspb.Message.toObjectList(msg.getComponentsList(),
    proto.bigblue.products.v0.BundleComponent.toObject, includeInstance),
    optionsList: jspb.Message.toObjectList(msg.getOptionsList(),
    proto.bigblue.products.v0.ProductOption.toObject, includeInstance),
    componentsUpdateTime: jspb.Message.getFieldWithDefault(msg, 5, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.Bundle}
 */
proto.bigblue.products.v0.Bundle.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.Bundle;
  return proto.bigblue.products.v0.Bundle.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.Bundle} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.Bundle}
 */
proto.bigblue.products.v0.Bundle.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setDisplayName(value);
      break;
    case 3:
      var value = new proto.bigblue.products.v0.BundleComponent;
      reader.readMessage(value,proto.bigblue.products.v0.BundleComponent.deserializeBinaryFromReader);
      msg.addComponents(value);
      break;
    case 4:
      var value = new proto.bigblue.products.v0.ProductOption;
      reader.readMessage(value,proto.bigblue.products.v0.ProductOption.deserializeBinaryFromReader);
      msg.addOptions(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setComponentsUpdateTime(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.Bundle.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.Bundle.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.Bundle} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.Bundle.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getDisplayName();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getComponentsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      3,
      f,
      proto.bigblue.products.v0.BundleComponent.serializeBinaryToWriter
    );
  }
  f = message.getOptionsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      4,
      f,
      proto.bigblue.products.v0.ProductOption.serializeBinaryToWriter
    );
  }
  f = message.getComponentsUpdateTime();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.bigblue.products.v0.Bundle.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.Bundle} returns this
 */
proto.bigblue.products.v0.Bundle.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string display_name = 2;
 * @return {string}
 */
proto.bigblue.products.v0.Bundle.prototype.getDisplayName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.Bundle} returns this
 */
proto.bigblue.products.v0.Bundle.prototype.setDisplayName = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * repeated BundleComponent components = 3;
 * @return {!Array<!proto.bigblue.products.v0.BundleComponent>}
 */
proto.bigblue.products.v0.Bundle.prototype.getComponentsList = function() {
  return /** @type{!Array<!proto.bigblue.products.v0.BundleComponent>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.bigblue.products.v0.BundleComponent, 3));
};


/**
 * @param {!Array<!proto.bigblue.products.v0.BundleComponent>} value
 * @return {!proto.bigblue.products.v0.Bundle} returns this
*/
proto.bigblue.products.v0.Bundle.prototype.setComponentsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 3, value);
};


/**
 * @param {!proto.bigblue.products.v0.BundleComponent=} opt_value
 * @param {number=} opt_index
 * @return {!proto.bigblue.products.v0.BundleComponent}
 */
proto.bigblue.products.v0.Bundle.prototype.addComponents = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 3, opt_value, proto.bigblue.products.v0.BundleComponent, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.bigblue.products.v0.Bundle} returns this
 */
proto.bigblue.products.v0.Bundle.prototype.clearComponentsList = function() {
  return this.setComponentsList([]);
};


/**
 * repeated ProductOption options = 4;
 * @return {!Array<!proto.bigblue.products.v0.ProductOption>}
 */
proto.bigblue.products.v0.Bundle.prototype.getOptionsList = function() {
  return /** @type{!Array<!proto.bigblue.products.v0.ProductOption>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.bigblue.products.v0.ProductOption, 4));
};


/**
 * @param {!Array<!proto.bigblue.products.v0.ProductOption>} value
 * @return {!proto.bigblue.products.v0.Bundle} returns this
*/
proto.bigblue.products.v0.Bundle.prototype.setOptionsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 4, value);
};


/**
 * @param {!proto.bigblue.products.v0.ProductOption=} opt_value
 * @param {number=} opt_index
 * @return {!proto.bigblue.products.v0.ProductOption}
 */
proto.bigblue.products.v0.Bundle.prototype.addOptions = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 4, opt_value, proto.bigblue.products.v0.ProductOption, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.bigblue.products.v0.Bundle} returns this
 */
proto.bigblue.products.v0.Bundle.prototype.clearOptionsList = function() {
  return this.setOptionsList([]);
};


/**
 * optional string components_update_time = 5;
 * @return {string}
 */
proto.bigblue.products.v0.Bundle.prototype.getComponentsUpdateTime = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.Bundle} returns this
 */
proto.bigblue.products.v0.Bundle.prototype.setComponentsUpdateTime = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.BundleComponent.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.BundleComponent.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.BundleComponent} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.BundleComponent.toObject = function(includeInstance, msg) {
  var f, obj = {
    product: jspb.Message.getFieldWithDefault(msg, 1, ""),
    quantity: jspb.Message.getFieldWithDefault(msg, 2, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.BundleComponent}
 */
proto.bigblue.products.v0.BundleComponent.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.BundleComponent;
  return proto.bigblue.products.v0.BundleComponent.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.BundleComponent} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.BundleComponent}
 */
proto.bigblue.products.v0.BundleComponent.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setProduct(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readUint32());
      msg.setQuantity(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.BundleComponent.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.BundleComponent.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.BundleComponent} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.BundleComponent.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getProduct();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getQuantity();
  if (f !== 0) {
    writer.writeUint32(
      2,
      f
    );
  }
};


/**
 * optional string product = 1;
 * @return {string}
 */
proto.bigblue.products.v0.BundleComponent.prototype.getProduct = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.BundleComponent} returns this
 */
proto.bigblue.products.v0.BundleComponent.prototype.setProduct = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional uint32 quantity = 2;
 * @return {number}
 */
proto.bigblue.products.v0.BundleComponent.prototype.getQuantity = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.bigblue.products.v0.BundleComponent} returns this
 */
proto.bigblue.products.v0.BundleComponent.prototype.setQuantity = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.BundleVersion.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.BundleVersion.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.BundleVersion} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.BundleVersion.toObject = function(includeInstance, msg) {
  var f, obj = {
    correlationId: jspb.Message.getFieldWithDefault(msg, 1, ""),
    createTime: jspb.Message.getFieldWithDefault(msg, 2, ""),
    hash: jspb.Message.getFieldWithDefault(msg, 3, 0),
    changeType: jspb.Message.getFieldWithDefault(msg, 4, 0),
    bundle: (f = msg.getBundle()) && proto.bigblue.products.v0.Bundle.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.BundleVersion}
 */
proto.bigblue.products.v0.BundleVersion.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.BundleVersion;
  return proto.bigblue.products.v0.BundleVersion.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.BundleVersion} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.BundleVersion}
 */
proto.bigblue.products.v0.BundleVersion.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setCorrelationId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setCreateTime(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readFixed64());
      msg.setHash(value);
      break;
    case 4:
      var value = /** @type {!proto.bigblue.common.ChangeType} */ (reader.readEnum());
      msg.setChangeType(value);
      break;
    case 5:
      var value = new proto.bigblue.products.v0.Bundle;
      reader.readMessage(value,proto.bigblue.products.v0.Bundle.deserializeBinaryFromReader);
      msg.setBundle(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.BundleVersion.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.BundleVersion.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.BundleVersion} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.BundleVersion.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getCorrelationId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getCreateTime();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getHash();
  if (f !== 0) {
    writer.writeFixed64(
      3,
      f
    );
  }
  f = message.getChangeType();
  if (f !== 0.0) {
    writer.writeEnum(
      4,
      f
    );
  }
  f = message.getBundle();
  if (f != null) {
    writer.writeMessage(
      5,
      f,
      proto.bigblue.products.v0.Bundle.serializeBinaryToWriter
    );
  }
};


/**
 * optional string correlation_id = 1;
 * @return {string}
 */
proto.bigblue.products.v0.BundleVersion.prototype.getCorrelationId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.BundleVersion} returns this
 */
proto.bigblue.products.v0.BundleVersion.prototype.setCorrelationId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string create_time = 2;
 * @return {string}
 */
proto.bigblue.products.v0.BundleVersion.prototype.getCreateTime = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.BundleVersion} returns this
 */
proto.bigblue.products.v0.BundleVersion.prototype.setCreateTime = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional fixed64 hash = 3;
 * @return {number}
 */
proto.bigblue.products.v0.BundleVersion.prototype.getHash = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {number} value
 * @return {!proto.bigblue.products.v0.BundleVersion} returns this
 */
proto.bigblue.products.v0.BundleVersion.prototype.setHash = function(value) {
  return jspb.Message.setProto3IntField(this, 3, value);
};


/**
 * optional bigblue.common.ChangeType change_type = 4;
 * @return {!proto.bigblue.common.ChangeType}
 */
proto.bigblue.products.v0.BundleVersion.prototype.getChangeType = function() {
  return /** @type {!proto.bigblue.common.ChangeType} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {!proto.bigblue.common.ChangeType} value
 * @return {!proto.bigblue.products.v0.BundleVersion} returns this
 */
proto.bigblue.products.v0.BundleVersion.prototype.setChangeType = function(value) {
  return jspb.Message.setProto3EnumField(this, 4, value);
};


/**
 * optional Bundle bundle = 5;
 * @return {?proto.bigblue.products.v0.Bundle}
 */
proto.bigblue.products.v0.BundleVersion.prototype.getBundle = function() {
  return /** @type{?proto.bigblue.products.v0.Bundle} */ (
    jspb.Message.getWrapperField(this, proto.bigblue.products.v0.Bundle, 5));
};


/**
 * @param {?proto.bigblue.products.v0.Bundle|undefined} value
 * @return {!proto.bigblue.products.v0.BundleVersion} returns this
*/
proto.bigblue.products.v0.BundleVersion.prototype.setBundle = function(value) {
  return jspb.Message.setWrapperField(this, 5, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.bigblue.products.v0.BundleVersion} returns this
 */
proto.bigblue.products.v0.BundleVersion.prototype.clearBundle = function() {
  return this.setBundle(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.bigblue.products.v0.BundleVersion.prototype.hasBundle = function() {
  return jspb.Message.getField(this, 5) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.ProductImage.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.ProductImage.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.ProductImage} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductImage.toObject = function(includeInstance, msg) {
  var f, obj = {
    sourceUrl: jspb.Message.getFieldWithDefault(msg, 1, ""),
    updateTime: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.ProductImage}
 */
proto.bigblue.products.v0.ProductImage.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.ProductImage;
  return proto.bigblue.products.v0.ProductImage.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.ProductImage} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.ProductImage}
 */
proto.bigblue.products.v0.ProductImage.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setSourceUrl(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setUpdateTime(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.ProductImage.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.ProductImage.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.ProductImage} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductImage.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getSourceUrl();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getUpdateTime();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional string source_url = 1;
 * @return {string}
 */
proto.bigblue.products.v0.ProductImage.prototype.getSourceUrl = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductImage} returns this
 */
proto.bigblue.products.v0.ProductImage.prototype.setSourceUrl = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string update_time = 2;
 * @return {string}
 */
proto.bigblue.products.v0.ProductImage.prototype.getUpdateTime = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductImage} returns this
 */
proto.bigblue.products.v0.ProductImage.prototype.setUpdateTime = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.ProductBuyerLocalizationOption.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.ProductBuyerLocalizationOption.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.ProductBuyerLocalizationOption} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductBuyerLocalizationOption.toObject = function(includeInstance, msg) {
  var f, obj = {
    displayName: jspb.Message.getFieldWithDefault(msg, 1, ""),
    value: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.ProductBuyerLocalizationOption}
 */
proto.bigblue.products.v0.ProductBuyerLocalizationOption.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.ProductBuyerLocalizationOption;
  return proto.bigblue.products.v0.ProductBuyerLocalizationOption.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.ProductBuyerLocalizationOption} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.ProductBuyerLocalizationOption}
 */
proto.bigblue.products.v0.ProductBuyerLocalizationOption.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setDisplayName(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setValue(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.ProductBuyerLocalizationOption.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.ProductBuyerLocalizationOption.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.ProductBuyerLocalizationOption} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductBuyerLocalizationOption.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getDisplayName();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getValue();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional string display_name = 1;
 * @return {string}
 */
proto.bigblue.products.v0.ProductBuyerLocalizationOption.prototype.getDisplayName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductBuyerLocalizationOption} returns this
 */
proto.bigblue.products.v0.ProductBuyerLocalizationOption.prototype.setDisplayName = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string value = 2;
 * @return {string}
 */
proto.bigblue.products.v0.ProductBuyerLocalizationOption.prototype.getValue = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductBuyerLocalizationOption} returns this
 */
proto.bigblue.products.v0.ProductBuyerLocalizationOption.prototype.setValue = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.bigblue.products.v0.ProductBuyerLocalization.repeatedFields_ = [4];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.ProductBuyerLocalization.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.ProductBuyerLocalization.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.ProductBuyerLocalization} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductBuyerLocalization.toObject = function(includeInstance, msg) {
  var f, obj = {
    product: jspb.Message.getFieldWithDefault(msg, 1, ""),
    language: jspb.Message.getFieldWithDefault(msg, 2, ""),
    displayName: jspb.Message.getFieldWithDefault(msg, 3, ""),
    optionsList: jspb.Message.toObjectList(msg.getOptionsList(),
    proto.bigblue.products.v0.ProductBuyerLocalizationOption.toObject, includeInstance),
    mainImageUrl: jspb.Message.getFieldWithDefault(msg, 5, ""),
    sourceShop: jspb.Message.getFieldWithDefault(msg, 6, ""),
    updateTime: jspb.Message.getFieldWithDefault(msg, 7, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.ProductBuyerLocalization}
 */
proto.bigblue.products.v0.ProductBuyerLocalization.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.ProductBuyerLocalization;
  return proto.bigblue.products.v0.ProductBuyerLocalization.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.ProductBuyerLocalization} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.ProductBuyerLocalization}
 */
proto.bigblue.products.v0.ProductBuyerLocalization.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setProduct(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setLanguage(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setDisplayName(value);
      break;
    case 4:
      var value = new proto.bigblue.products.v0.ProductBuyerLocalizationOption;
      reader.readMessage(value,proto.bigblue.products.v0.ProductBuyerLocalizationOption.deserializeBinaryFromReader);
      msg.addOptions(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setMainImageUrl(value);
      break;
    case 6:
      var value = /** @type {string} */ (reader.readString());
      msg.setSourceShop(value);
      break;
    case 7:
      var value = /** @type {string} */ (reader.readString());
      msg.setUpdateTime(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.ProductBuyerLocalization.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.ProductBuyerLocalization.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.ProductBuyerLocalization} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.ProductBuyerLocalization.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getProduct();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getLanguage();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getDisplayName();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getOptionsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      4,
      f,
      proto.bigblue.products.v0.ProductBuyerLocalizationOption.serializeBinaryToWriter
    );
  }
  f = message.getMainImageUrl();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
  f = message.getSourceShop();
  if (f.length > 0) {
    writer.writeString(
      6,
      f
    );
  }
  f = message.getUpdateTime();
  if (f.length > 0) {
    writer.writeString(
      7,
      f
    );
  }
};


/**
 * optional string product = 1;
 * @return {string}
 */
proto.bigblue.products.v0.ProductBuyerLocalization.prototype.getProduct = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductBuyerLocalization} returns this
 */
proto.bigblue.products.v0.ProductBuyerLocalization.prototype.setProduct = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string language = 2;
 * @return {string}
 */
proto.bigblue.products.v0.ProductBuyerLocalization.prototype.getLanguage = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductBuyerLocalization} returns this
 */
proto.bigblue.products.v0.ProductBuyerLocalization.prototype.setLanguage = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string display_name = 3;
 * @return {string}
 */
proto.bigblue.products.v0.ProductBuyerLocalization.prototype.getDisplayName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductBuyerLocalization} returns this
 */
proto.bigblue.products.v0.ProductBuyerLocalization.prototype.setDisplayName = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * repeated ProductBuyerLocalizationOption options = 4;
 * @return {!Array<!proto.bigblue.products.v0.ProductBuyerLocalizationOption>}
 */
proto.bigblue.products.v0.ProductBuyerLocalization.prototype.getOptionsList = function() {
  return /** @type{!Array<!proto.bigblue.products.v0.ProductBuyerLocalizationOption>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.bigblue.products.v0.ProductBuyerLocalizationOption, 4));
};


/**
 * @param {!Array<!proto.bigblue.products.v0.ProductBuyerLocalizationOption>} value
 * @return {!proto.bigblue.products.v0.ProductBuyerLocalization} returns this
*/
proto.bigblue.products.v0.ProductBuyerLocalization.prototype.setOptionsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 4, value);
};


/**
 * @param {!proto.bigblue.products.v0.ProductBuyerLocalizationOption=} opt_value
 * @param {number=} opt_index
 * @return {!proto.bigblue.products.v0.ProductBuyerLocalizationOption}
 */
proto.bigblue.products.v0.ProductBuyerLocalization.prototype.addOptions = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 4, opt_value, proto.bigblue.products.v0.ProductBuyerLocalizationOption, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.bigblue.products.v0.ProductBuyerLocalization} returns this
 */
proto.bigblue.products.v0.ProductBuyerLocalization.prototype.clearOptionsList = function() {
  return this.setOptionsList([]);
};


/**
 * optional string main_image_url = 5;
 * @return {string}
 */
proto.bigblue.products.v0.ProductBuyerLocalization.prototype.getMainImageUrl = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductBuyerLocalization} returns this
 */
proto.bigblue.products.v0.ProductBuyerLocalization.prototype.setMainImageUrl = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};


/**
 * optional string source_shop = 6;
 * @return {string}
 */
proto.bigblue.products.v0.ProductBuyerLocalization.prototype.getSourceShop = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 6, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductBuyerLocalization} returns this
 */
proto.bigblue.products.v0.ProductBuyerLocalization.prototype.setSourceShop = function(value) {
  return jspb.Message.setProto3StringField(this, 6, value);
};


/**
 * optional string update_time = 7;
 * @return {string}
 */
proto.bigblue.products.v0.ProductBuyerLocalization.prototype.getUpdateTime = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 7, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.ProductBuyerLocalization} returns this
 */
proto.bigblue.products.v0.ProductBuyerLocalization.prototype.setUpdateTime = function(value) {
  return jspb.Message.setProto3StringField(this, 7, value);
};


/**
 * @enum {number}
 */
proto.bigblue.products.v0.StorageType = {
  STORAGE_TYPE_DEFAULT: 0,
  HANGING_STORAGE_SHORT: 1,
  HANGING_STORAGE_LONG: 2
};

/**
 * @enum {number}
 */
proto.bigblue.products.v0.HandlingType = {
  HANDLING_TYPE_DEFAULT: 0,
  HANDLING_MANUAL: 1
};

/**
 * @enum {number}
 */
proto.bigblue.products.v0.Packaging = {
  PACKAGING_UNKNOWN: 0,
  SHIP_READY: 1
};

/**
 * @enum {number}
 */
proto.bigblue.products.v0.ProductBuyerLocalizationShopPriority = {
  PRODUCT_BUYER_LOCALIZATION_SHOP_PRIORITY_UNSPECIFIED: 0,
  SHOP_PRIORITY_MAIN: 1,
  SHOP_PRIORITY_FALLBACK: 2,
  SHOP_PRIORITY_IGNORED: 3
};

goog.object.extend(exports, proto.bigblue.products.v0);
