// proto-ts-products/v0/warehouseservice_pb.js
// source: bigblue/products/v0/warehouseservice.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global =
    (typeof globalThis !== 'undefined' && globalThis) ||
    (typeof window !== 'undefined' && window) ||
    (typeof global !== 'undefined' && global) ||
    (typeof self !== 'undefined' && self) ||
    (function () { return this; }).call(null) ||
    Function('return this')();

var bigblue_products_v0_packunit_pb = require('./packunit_pb.js');
goog.object.extend(proto, bigblue_products_v0_packunit_pb);
var bigblue_products_v0_product_pb = require('./product_pb.js');
goog.object.extend(proto, bigblue_products_v0_product_pb);
var google_protobuf_empty_pb = require('google-protobuf/google/protobuf/empty_pb.js');
goog.object.extend(proto, google_protobuf_empty_pb);
goog.exportSymbol('proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest', null, global);
goog.exportSymbol('proto.bigblue.products.v0.WarehouseGetProductRequest', null, global);
goog.exportSymbol('proto.bigblue.products.v0.WarehouseListMarketingInsertProductsRequest', null, global);
goog.exportSymbol('proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse', null, global);
goog.exportSymbol('proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest', null, global);
goog.exportSymbol('proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.WarehouseGetProductRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.bigblue.products.v0.WarehouseGetProductRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.WarehouseGetProductRequest.displayName = 'proto.bigblue.products.v0.WarehouseGetProductRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest.repeatedFields_, null);
};
goog.inherits(proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest.displayName = 'proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.WarehouseListMarketingInsertProductsRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.bigblue.products.v0.WarehouseListMarketingInsertProductsRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.WarehouseListMarketingInsertProductsRequest.displayName = 'proto.bigblue.products.v0.WarehouseListMarketingInsertProductsRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse.repeatedFields_, null);
};
goog.inherits(proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse.displayName = 'proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest.displayName = 'proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse.repeatedFields_, null);
};
goog.inherits(proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse.displayName = 'proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.WarehouseGetProductRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.WarehouseGetProductRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.WarehouseGetProductRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.WarehouseGetProductRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.WarehouseGetProductRequest}
 */
proto.bigblue.products.v0.WarehouseGetProductRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.WarehouseGetProductRequest;
  return proto.bigblue.products.v0.WarehouseGetProductRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.WarehouseGetProductRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.WarehouseGetProductRequest}
 */
proto.bigblue.products.v0.WarehouseGetProductRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.WarehouseGetProductRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.WarehouseGetProductRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.WarehouseGetProductRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.WarehouseGetProductRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.bigblue.products.v0.WarehouseGetProductRequest.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.WarehouseGetProductRequest} returns this
 */
proto.bigblue.products.v0.WarehouseGetProductRequest.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    productsList: jspb.Message.toObjectList(msg.getProductsList(),
    bigblue_products_v0_product_pb.ProductForm.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest}
 */
proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest;
  return proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest}
 */
proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new bigblue_products_v0_product_pb.ProductForm;
      reader.readMessage(value,bigblue_products_v0_product_pb.ProductForm.deserializeBinaryFromReader);
      msg.addProducts(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getProductsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      1,
      f,
      bigblue_products_v0_product_pb.ProductForm.serializeBinaryToWriter
    );
  }
};


/**
 * repeated ProductForm products = 1;
 * @return {!Array<!proto.bigblue.products.v0.ProductForm>}
 */
proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest.prototype.getProductsList = function() {
  return /** @type{!Array<!proto.bigblue.products.v0.ProductForm>} */ (
    jspb.Message.getRepeatedWrapperField(this, bigblue_products_v0_product_pb.ProductForm, 1));
};


/**
 * @param {!Array<!proto.bigblue.products.v0.ProductForm>} value
 * @return {!proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest} returns this
*/
proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest.prototype.setProductsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 1, value);
};


/**
 * @param {!proto.bigblue.products.v0.ProductForm=} opt_value
 * @param {number=} opt_index
 * @return {!proto.bigblue.products.v0.ProductForm}
 */
proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest.prototype.addProducts = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 1, opt_value, proto.bigblue.products.v0.ProductForm, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest} returns this
 */
proto.bigblue.products.v0.WarehouseBatchUpdateProductMeasuresRequest.prototype.clearProductsList = function() {
  return this.setProductsList([]);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.WarehouseListMarketingInsertProductsRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.WarehouseListMarketingInsertProductsRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.WarehouseListMarketingInsertProductsRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.WarehouseListMarketingInsertProductsRequest.toObject = function(includeInstance, msg) {
  var f, obj = {

  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.WarehouseListMarketingInsertProductsRequest}
 */
proto.bigblue.products.v0.WarehouseListMarketingInsertProductsRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.WarehouseListMarketingInsertProductsRequest;
  return proto.bigblue.products.v0.WarehouseListMarketingInsertProductsRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.WarehouseListMarketingInsertProductsRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.WarehouseListMarketingInsertProductsRequest}
 */
proto.bigblue.products.v0.WarehouseListMarketingInsertProductsRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.WarehouseListMarketingInsertProductsRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.WarehouseListMarketingInsertProductsRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.WarehouseListMarketingInsertProductsRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.WarehouseListMarketingInsertProductsRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    productsList: jspb.Message.toObjectList(msg.getProductsList(),
    bigblue_products_v0_product_pb.Product.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse}
 */
proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse;
  return proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse}
 */
proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new bigblue_products_v0_product_pb.Product;
      reader.readMessage(value,bigblue_products_v0_product_pb.Product.deserializeBinaryFromReader);
      msg.addProducts(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getProductsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      1,
      f,
      bigblue_products_v0_product_pb.Product.serializeBinaryToWriter
    );
  }
};


/**
 * repeated Product products = 1;
 * @return {!Array<!proto.bigblue.products.v0.Product>}
 */
proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse.prototype.getProductsList = function() {
  return /** @type{!Array<!proto.bigblue.products.v0.Product>} */ (
    jspb.Message.getRepeatedWrapperField(this, bigblue_products_v0_product_pb.Product, 1));
};


/**
 * @param {!Array<!proto.bigblue.products.v0.Product>} value
 * @return {!proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse} returns this
*/
proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse.prototype.setProductsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 1, value);
};


/**
 * @param {!proto.bigblue.products.v0.Product=} opt_value
 * @param {number=} opt_index
 * @return {!proto.bigblue.products.v0.Product}
 */
proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse.prototype.addProducts = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 1, opt_value, proto.bigblue.products.v0.Product, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse} returns this
 */
proto.bigblue.products.v0.WarehouseListMarketingInsertProductsResponse.prototype.clearProductsList = function() {
  return this.setProductsList([]);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    product: jspb.Message.getFieldWithDefault(msg, 1, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest}
 */
proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest;
  return proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest}
 */
proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setProduct(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getProduct();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
};


/**
 * optional string product = 1;
 * @return {string}
 */
proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest.prototype.getProduct = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest} returns this
 */
proto.bigblue.products.v0.WarehouseListProductPackUnitsRequest.prototype.setProduct = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    productPackUnitsList: jspb.Message.toObjectList(msg.getProductPackUnitsList(),
    bigblue_products_v0_packunit_pb.ProductPackUnit.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse}
 */
proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse;
  return proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse}
 */
proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new bigblue_products_v0_packunit_pb.ProductPackUnit;
      reader.readMessage(value,bigblue_products_v0_packunit_pb.ProductPackUnit.deserializeBinaryFromReader);
      msg.addProductPackUnits(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getProductPackUnitsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      1,
      f,
      bigblue_products_v0_packunit_pb.ProductPackUnit.serializeBinaryToWriter
    );
  }
};


/**
 * repeated ProductPackUnit product_pack_units = 1;
 * @return {!Array<!proto.bigblue.products.v0.ProductPackUnit>}
 */
proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse.prototype.getProductPackUnitsList = function() {
  return /** @type{!Array<!proto.bigblue.products.v0.ProductPackUnit>} */ (
    jspb.Message.getRepeatedWrapperField(this, bigblue_products_v0_packunit_pb.ProductPackUnit, 1));
};


/**
 * @param {!Array<!proto.bigblue.products.v0.ProductPackUnit>} value
 * @return {!proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse} returns this
*/
proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse.prototype.setProductPackUnitsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 1, value);
};


/**
 * @param {!proto.bigblue.products.v0.ProductPackUnit=} opt_value
 * @param {number=} opt_index
 * @return {!proto.bigblue.products.v0.ProductPackUnit}
 */
proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse.prototype.addProductPackUnits = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 1, opt_value, proto.bigblue.products.v0.ProductPackUnit, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse} returns this
 */
proto.bigblue.products.v0.WarehouseListProductPackUnitsResponse.prototype.clearProductPackUnitsList = function() {
  return this.setProductPackUnitsList([]);
};


goog.object.extend(exports, proto.bigblue.products.v0);
